\documentclass[fleqn]{wlscirep}
\usepackage[utf8]{inputenc}
\usepackage{lipsum}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{amsfonts}
\usepackage{siunitx}
\usepackage{physics}
\usepackage{stmaryrd} % double brackets for jump operator
\usepackage{nameref}
\usepackage{booktabs}
\usepackage{bm}
\usepackage{amsopn}
\usepackage[skip=10pt]{caption}
\usepackage{color}
\usepackage{xcolor,colortbl}
\usepackage[nameinlink, capitalise]{cleveref}
\usepackage{graphicx}
\usepackage{tkz-euclide} % draw tikz pictures
\usepackage{subcaption}
\usepackage{tablefootnote} % for table footnotes
\usepackage{float} % figure placement

% Bibliography tools
\usepackage{csquotes}

% Set some cref options
\newcommand{\crefrangeconjunction}{--} % Change crefrange from "to" to dashed

%% Some new commands
% Math
\newcommand{\cbar}{\overline{c}}

% Partial derivatives
\newcommand{\pdifft}[1]{\frac{\partial  #1}{\partial t}}
\newcommand{\pdiffx}[1]{\frac{\partial #1}{\partial x}}
\newcommand{\pdiffy}[1]{\frac{\partial #1}{\partial y}}
\newcommand{\pdiffxi}[1]{\frac{\partial #1}{\partial x_i}}
\newcommand{\pdiffxj}[1]{\frac{\partial #1}{\partial x_j}}
\newcommand{\secpdiffx}[1]{\frac{\partial^2 #1}{\partial x^2}}
\newcommand{\secpdiffxi}[1]{\frac{\partial^2 #1}{\partial x_i \partial x_i}}
\newcommand{\secpdiffxj}[1]{\frac{\partial^2 #1}{\partial x_j \partial x_j}}

% Norms
\newcommand{\normltwo}[1]{{ \vert\vert#1\vert\vert}_{L^2}}
\newcommand{\normltwovec}[1]{{ \vert\vert#1\vert\vert}_{\mathbf{L}^2}}
\newcommand{\normlinf}[1]{{\vert\vert#1\vert\vert}_{L^{\infty}}}
\newcommand{\Hnorm}[1]{\norm{#1}_{\mathbf{H}^1}}
\newcommand{\Hdivnorm}[1]{\norm{#1}_{\mathbf{H}(\mathrm{div};\Omega)}}

% Integrals
\newcommand{\dx}{\, \mathrm d\bm{x}}
\newcommand{\intO}[1]{\int_{\Omega}#1 \, \mathrm d\bm{x}}
\newcommand{\intG}[1]{\int_{\Gamma}#1 \, \mathrm ds}
\newcommand{\intF}[1]{\int_{\mathcal{F}_k}#1 \, \mathrm dS}
\newcommand{\intGnf}[1]{\int_{\Gamma_{\mathrm{nf}}}#1 \, \mathrm ds}
\newcommand{\intGinj}[1]{\int_{\Gamma_{\mathrm{inj}}}#1 \, \mathrm ds}
\newcommand{\intGin}[1]{\int_{\Gamma_{\mathrm{in}}}#1 \, \mathrm ds}
\newcommand{\intGout}[1]{\int_{\Gamma_{\mathrm{out}}}#1 \, \mathrm ds}
\newcommand{\intGinjout}[1]{\int_{\Gamma_{\mathrm{inj}}\cup\Gamma_{\mathrm{out}}}#1 \, \mathrm ds}
\newcommand{\intGs}[1]{\int_{\Gamma_s}#1 \, \mathrm ds}
\newcommand{\intGc}[1]{\int_{\Gamma_c}#1 \, \mathrm ds}
\newcommand{\intGp}[1]{\int_{\Gamma_p}#1 \, \mathrm ds}

% Domain and boundaries
\newcommand{\Gs}{\Gamma_{s}}
\newcommand{\Gc}{\Gamma_{c}}
\newcommand{\Gp}{\Gamma_{p}}
\newcommand{\Ginj}{\Gamma_{\mathrm{inj}}}
\newcommand{\Gnf}{\Gamma_{\mathrm{nf}}}
\newcommand{\Gin}{\Gamma_{\mathrm{in}}}
\newcommand{\Gout}{\Gamma_{\mathrm{out}}}

% Discontinuity operators
\newcommand{\avg}[1]{\{#1\}}
\newcommand{\jump}[1]{\llbracket#1\rrbracket}

% Bold faced math characters
\newcommand{\ff}{\mathbf{f}}
\newcommand{\nn}{\mathbf{n}}
\newcommand{\rr}{\mathbf{r}}
\newcommand{\uu}{\mathbf{u}}
\newcommand{\vv}{\mathbf{v}}
\newcommand{\ww}{\mathbf{w}}
\newcommand{\xx}{\bm{x}}
\newcommand{\JJ}{\mathbf{J}}
\newcommand{\VV}{\mathbf{V}}
\newcommand{\bsig}{\bm{\sigma}}
\newcommand{\bsigpar}{\hat{\bsig}_{\parallel}}
\newcommand{\bsigperp}{\hat{\bsig}_{\perp}}
\newcommand{\beps}{\bm{\varepsilon}}
\newcommand{\btau}{\bm{\tau}}

% Comments
\newcommand{\lyng}[1]{\textcolor{blue}{#1}}
\newcommand{\mer}[1]{\textcolor{magenta}{#1}}
\newcommand{\mk}[1]{\textcolor{olive}{MK: #1}}

\newcommand{\fixme}[1]{\textcolor{red}{#1}}

% Title
\title{The role of advection versus diffusion in intraventricular transport}
\author[1,*]{Halvor Herlyng}
\author[1]{Ada J.~Ellingsrud}
\author[1]{Miroslav Kuchta} 
\author[X]{Inyoung Jeong} % OK?
\author[1,3,+]{Marie E.~Rognes} % OK order?
\author[2,+, *]{Nathalie Jurisch-Yaksi}
\affil[1]{Department for Numerical Analysis and Scientific Computing, Simula Research Laboratory, Oslo, Norway}
\affil[2]{Department of Clinical and Molecular Medicine, Norwegian University of Science and Technology, Trondheim, Norway}
\affil[3]{K.~G.~Jebsen Center for Brain Fluid Research}
\affil[*]{\href{mailto:hherlyng@simula.no}{hherlyng@simula.no} \href{mailto:nathalie.jurisch-yaksi@ntnu.no}{nathalie.jurisch-yaksi@ntnu.no}}
\affil[+]{these authors contributed equally to this work}
% Optional PDF information
\ifpdf
\hypersetup{
  pdftitle={zebrafish-cilia-csf},
  pdfauthor={H.~Herlyng~\emph{et al.}}
}
\fi

\begin{abstract}
    This paper presents ...
    %Flow of cerebrospinal fluid in zebrafish brain ventricles is studied in this work. Motivated by experimental observations, we consider motile cilia and cardiac beating as the driving flow mechanisms.
    %A mathematical model of the flow is developed and coupled with a transport model to study the transport of molecules inside the brain ventricles. 
    %The Stokes equations are used to model the flow, with the motile cilia acting as a force on the fluid through a stress boundary condition that allows slip at the wall. 
    %Transport is modeled with an advection-diffusion equation coupled to the velocity field obtained from solving the Stokes equations.
    %The flow and transport models are discretized with finite element methods. For the Stokes equations we employ a non-conforming discretization based on elements of the Brezzi-Douglas-Marini family. 
    %The advection-diffusion equation is discretized with discontinuous Galerkin elements. 
    %We implement and solve the discretized equations numerically to simulate flow and transport in a zebrafish brain ventricles geometry obtained with medical imaging.
\end{abstract}
\begin{document}
\maketitle
% Introduction:
% 1. What are cilia and why are they important?
% 2. Previous studies of cilia flow
% 3. What problem do we want to solve
% 4. The model and work done in this paper
\section*{Introduction}
Brain development and function depend on numerous regulatory processes,
including interactions with cerebrospinal fluid (CSF).
CSF is a clear liquid produced by the choroid plexus, which is a secretory
epithelium located within the brain ventricles.
Following production, CSF flows through the ventricular system and
subarachnoid space, where it regulates brain development and
homeostasis by transporting growth factors, nutrients, neurotransmitters,
waste products and other molecules and metabolites. 

To achieve its function, CSF needs to circulate throughout the brain.
In fact, CSF disturbances can univocally lead to brain disorders,
such as hydrocephalus where accumulation of CSF requires emergency surgical treatment.
CSF movement occurs through multiple physiological processes, including the action of
motile cilia on the ventricular wall,
CSF secretion by the choroid plexus and pressure gradients generated by the
cardiac cycle, respiration, and bodily movement.
Motile cilia are microscopic hair-like structures that beat in a metachronal
wave pattern to generate flow pattern. Due to their microscopic size, ranging from 6 to 15 microns,
they contribute to fluid flow primarily at the surface of the brain ventricles.
In contrast, pressure gradients generate fluid flow further from the ventricular
walls and along the perivascular space surrounding the major blood vessels. 

To date, it remains poorly understood how cilia contribute to the overall fluid
flow and solute transport in the brain.
These questions have been especially challenging to tackle due to technical limitations.
Notably, cilia are very small and beat at high frequencies ranging from 10 to 30 Hz,
while other fluid dynamic processes in the brain are slower and span larger three-dimensional volumes.
Most studies until now have analyzed one process at a time, especially in mammals.
For instance, multiple studies have convincingly shown that cilia generate complex
flow patterns at the surface of the brain ventricles.
However, since all these studies have been performed on brain explants, they failed
to identify the role of pressure gradients associated with respiration and heart pulsations in CSF transport.
Similarly, while MRI technology can measure CSF movement in humans, it cannot detect
the fast movements of cilia or understand their impact in the large human brain.
Consequently, we lack a comprehensive understanding of how cilia contribute to CSF
movement and solute transport, and whether cilia dysfunction will alter CSF distribution
and lead to brain disorders.
This is particularly important since ventricular enlargement or hydrocephalus is
commonly observed in animal models with motile cilia deficiency, potentially affecting
brain activity and behavior.

The technical limitations of studying cilia-mediated flow and transport opens the
possibility of mathematical modeling and computational investigations. Cilia have been studied previously,
with numerous studies on the interaction between single or several cilia and surrounding
fluids~\cite{Guo2020SimulatingGeometries, Ruvalcaba2021NumericalTree, Smith2009MathematicalFluids,
Cui2019NumericalMethod, Cui2022AFlow}.
Others considered applications such as ciliary transport of mucous in
airways~\cite{Fulford1986Muco-ciliaryLung, Ramirez-SanJuan2020Multi-scaleArrays},
propulsion and locomotion of ciliated organisms~\cite{BLAKE1974MechanicsMotion,
Jahn1972LocomotionProtozoa, Brennen1977FluidFlagella},
coral reefs~\cite{Pacherres2022CiliaryProduction}, and the brain~\cite{Siyahhan2014FlowVentricles,
Yoshida2022EffectVentricles, Salman2022ComputationalEmbryo, Thouvenin2020OriginCanal}.
These studies have provided valuable insights, showing that cilia lend themselves to simulation studies.

In this study, we combined computational and experimental approaches to investigate
the role of motile cilia in CSF flow and solute transport.
First, we computed flow patterns and intraventricular transport \emph{in silico}. To
this end, we used the geometry of the zebrafish larval brain ventricles,
for which we have detailed knowledge of cilia properties and CSF movement velocities
and frequencies. Notably, this early developmental stage allows us to simplify our model
by omitting CSF secretion, which starts at later stages, while still being well-conserved with mammals.
Next, we validated our computation model with experimental data by monitoring
the dynamics of a photoconvertible fluorescent protein expressed within the brain ventricles.
Our results show that diffusion plays a major role in the transport of small solutes,
whereas cilia majorly contribute to the movement of larger particles.
We also found that the location of cilia, as well as the ventricular geometry, greatly
impact solute distribution. Altogether, our work presents a computational
framework together with new concepts of how cilia
collaborate with other factors to distribute molecules and
particles within the brain ventricles. 

\section*{Methods}
\subsection*{Image-based computational geometries of zebrafish brain ventricles}
To model flow and transport in zebrafish larval brain ventricles, we generated a
3D representation of the brain ventricular walls using confocal imaging data of a zebrafish embryo
injected intraventricularly with a 70 kDa dye at 2 dpf~\cite{Olstad2019CiliaryDevelopment}. 
From the imaging data, we constructed a computational mesh of the ventricles (\Cref{fig:figure1}d).
The interior volume was meshed with fTetWild~\cite{Hu2020FastWild}. 
The \emph{standard} mesh consisted of 132 134 tetrahedral cells and had a
maximal (minimal) edge length of 14.9 $\mu$m (4.68 $\mu$m). 
At 2 dpf, the ventricular system consists mainly of 3 cavities: the telencephalic (anterior),
the di-/mesencephalic (middle) and the rhombencephalic (posterior) ventricles, which are connected by ducts. 
Parts of the ventricular surfaces were marked to be lined with
motile cilia~\cite{Olstad2019CiliaryDevelopment} (\Cref{fig:figure1}d, magenta markers).
\begin{figure}%[hbt!]
    \centering
    \includegraphics[width=\linewidth]{graphics/figure1.png}
    \caption{\lyng{Experimenting with velocity colorbars, thoughts on the
    two options? Or have two separate as it is now?}}
    \label{fig:figure1}
\end{figure}

The morphology and geometry of the brain ventricles vary across zebrafish
individuals, and under physiological and pathological conditions. 
Motivated by an interest in how geometry impacts solute distribution,
we also consider four variations in the ventricular geometry: by shrinking the fore-mid brain connection, 
the middle ventricle, the mid-hind brain connection, and all three simultaneously. 
We used Blender~\cite{Community2018BlenderPackage} to modify the brain ventricles geometry.
\mer{MER: More specifics (i.e.~1 sentence and/or inserted info) here on how much these were shrunk and how.}

\subsection*{Computational CSF dynamics}
The beating motion of the cilia generates CSF flow with steady rotational
structures~\cite{Olstad2019CiliaryDevelopment}
within the brain ventricles (\Cref{fig:figure1}a--c).
In addition, cardiac pulsations induces pulsatile CSF flow.
We model this flow of CSF in the brain ventricles by the time-dependent,
incompressible Stokes equations which read as follows.
Find the CSF velocity $\uu = \uu(\xx, t) = (u_x(\xx, t), u_y(\xx, t), u_z(\xx, t))$
and the CSF pressure $p = p(\xx, t)$ for $\xx \in \Omega$ and time $t>0$, such that
\begin{subequations}
    \begin{align}
      \rho \pdifft{\uu} - \nabla \cdot \bsig &= \mathbf{0}
      \quad \text{in } \Omega,\\
      \nabla \cdot \uu &= 0
      \quad  \text{in } \Omega,
  \end{align}
  \label{eq:stokes_eqs}% 
\end{subequations}%
where the stress tensor is defined as $\bsig(\uu, p) = 2\mu\beps(\uu) - p\mathbf{I}$
and $\beps(\uu) = \frac{1}{2}\left(\nabla \uu + (\nabla\uu)^T\right)$ is the strain-rate tensor. 
Bold-face characters denote vectors or tensors, and $\mathbf{I}$ is the identity tensor in three dimensions.
Furthermore, $\rho=1000 \ \mathrm{kg/m^3}$ is the CSF density
and $\mu=6.97\times 10^{-4} \ \mathrm{Pa\cdot s}$ is the dynamic viscosity
of the CSF~\cite{Bloomfield1998EffectsFluid}.
We introduce the tangential traction $\bsigpar$ as the
tangential component of the traction $\hat{\bsig}=\bsig\nn$:
\begin{equation*}
    \bsigpar = P_{\nn}(\hat{\bsig}) = (\mathbf{I} - \nn\otimes\nn)\hat{\bsig},
\end{equation*}
where $\nn$ is the outer unit normal of the boundary surface $\Gamma = \partial\Omega$,
and $P_{\nn}(\rr)$ is the tangential projection of a vector $\rr$ onto $\Gamma$.
We consider three types of boundary conditions:
(i) \emph{tangential traction} (no flow normal to the boundary
$\uu \cdot \nn = 0$ with a force $\btau$ applied tangentially on the boundary $\bsigpar = \btau$),
(ii) \emph{normal pressure} (a pressure $\Tilde{p}$ applied in
the normal direction: $(\mu\nabla\uu - p\mathbf{I})\nn = \Tilde{p}(t)\nn$),
and (iii) \emph{free slip} (no flow normal to the boundary
$\uu \cdot \nn = 0$ and no tangential forces $\bsigpar = {\bf 0}$).
Unless noted otherwise, the free-slip condition is applied on the
boundary. We assume that the system starts at rest: $\uu(\xx, t=0) = \mathbf{0}$.

To study the impact of the motile cilia and the cardiac pulsations on the CSF flow,
we consider the following computational flow scenarios,
each predicting a CSF flow velocity $\uu$ and CSF pressure $p$.
\begin{itemize}
    \item A \emph{baseline} flow model representing a wild-type zebrafish including 
    contributions both from the cilia forces and the cardiac pulsations.
    In this scenario, we apply the tangential traction $\btau$ on the cilia boundary
    $\Gc$ and prescribe the normal pressure $\Tilde{p}$ on the anterior/posterior boundary $\Gp$. % Model 0

    \item A \emph{cilia-only/no-cardiac} flow model including contributions from the
    cilia forces but no cardiac pulsations.
    Here, we apply the tangential traction $\btau$ on the cilia boundary $\Gc$.
    In the absence of cardiac flow, the anterior/posterior boundary $\Gp=\emptyset$. % Model I

    \item A \emph{cardiac-only/no-cilia} flow model including contributions from the
    cardiac pulsations but no cilia forces.
    Here, we set the tangential traction to zero on the cilia boundary
    $\Gc$, while prescribing the normal pressure $\Tilde{p}$ on the
    anterior/posterior boundary $\Gp$.  % Model II
\end{itemize}

\subsection*{Cilia-driven flow}
On the ciliated regions of the ventricle walls $\Gc$ (\Cref{fig:figure1}d, magenta markers),
we impose a constant-in-time tangential traction $\bsigpar=\btau$ to represent the net forces
of the cilia acting on the CSF.
We set $\btau(\xx, \rr) = \tau \lambda(\xx) P_{\nn}(\rr)$, where the function $\lambda(\xx)$
and the sign of the vector $\rr=\pm(1, 0, 1)$ varies depending on the
cilia region (see~\Cref{subsec:appendixA1}).
Both $\rr$ and the value $\tau=6.5\times 10^{-4}$ Pa were chosen by 
model calibration using experimental data~\cite{Olstad2019CiliaryDevelopment}.
The data indicate speeds proximate to the ventricular walls of $27.4 \pm 5.4$ \textmu m/s and
$2.6 \pm 0.6$ \textmu m/s in the dorsal and ventral regions of the middle ventricle, respectively.
In the anterior ventricle, the speeds were estimated at $4.7 \pm 1.3$ \textmu m/s.
We calibrated $\tau$ heuristically by comparing these experimental velocity values
with the simulated, maximum velocity magnitude.

\subsection*{Flow induced by cardiac pulsations}\label{subsec:pressure_bcs}
To drive a cardiac-induced pulsatile flow inside the brain ventricles, we set $\Tilde{p}(t)=0$ and
$\Tilde{p}(t)=-A\sin{\omega t}$ on anterior and posterior parts, respectively, of the
boundary $\Gp$~(\Cref{fig:figure1}d, blue markers).
Based on a measured cardiac frequency of $f=2.22$ Hz~\cite{Olstad2019CiliaryDevelopment},
the angular cardiac frequency $\omega=2\pi f=6.97$ rad/s.
The amplitude $A=1.5\times 10^{-3}$ Pa determines the magnitude of the pulsatile flow,
and was calibrated based on experimental data~\cite{Olstad2019CiliaryDevelopment}.
Similar to $\tau$, the amplitude $A$ was chosen by comparing experimental and simulated velocities.
Particle tracking data of pulsatile motion was used to determine the contributions of the
cardiac pulsations to the velocity magnitude,
as compared to the effect of cilia motion on the flow (see~\Cref{sec:appC_particle_transport}).

\subsection*{Computational model of solute transport after photoconversion}
We simulate transport of photoconverted proteins within the brain ventricles by
modeling transport of a solute with concentration $c(\xx, t)$ for $\xx\in\Omega$ and
time $t > 0$ by the advection-diffusion equation
\begin{equation}
    \pdifft{c} + \nabla\cdot\JJ = 0 \quad \text{in } \Omega,
    \label{eq:adv_diff_strong}
\end{equation}
where the total concentration flux $\mathbf{J}$ is assumed to consist of an advective and a diffusive flux:
\begin{equation*}
    \JJ = c\uu - D\nabla c .
\end{equation*}
Here, $\uu$ is the CSF velocity field, governed by the Stokes equations~\eqref{eq:stokes_eqs},
and $D$ is the molecular diffusion coefficient of the solute.
Initially, we set $c(\xx, 0)=0$ for all $\xx\in\Omega$.
The ventricular walls are assumed to be impermeable (admitting \emph{no flux}: $\JJ\cdot\nn=0$),
except at the anterior/posterior boundary $\Gamma_p$.
At the anterior/posterior boundary, we permit the solute concentration to be advected into
and out of the domain in a manner that ensures conservation of mass
(see~\Cref{subsec:adv_diff_eq_weak_form} for details).

To computationally represent protein photoconversion in a region
$\Omega_{\mathrm{pc}} \subset \Omega$ (corresponding to ROI 1),
we prescribe a given time-dependent value to the solute concentration in this region. 
Mimicking the experimental protocol, we consider the dorsal part of the
diencephalic ventricle as the subdomain $\Omega_{\mathrm{pc}}$, and set
\begin{equation}
  c(\xx, t)=\frac{\log{(1+t/a)}}{\log{(1+T/a)}},
  \quad \xx \in \Omega_{\mathrm{pc}}, \quad t > 0.
\label{eq:photoconversion_curve}
\end{equation}
The value $a=65$ was chosen to fit the fluorescence intensity curve
observed in the physical photoconversion experiments.

\subsection*{Estimation of diffusion coefficients via the Stokes-Einstein relation}
Motivated by an interest in molecules that are significant in neural development,
we study the transport of molecules resembling the Dendra2 Fluorescent Protein (Dendra2)
used in this study, the Starmaker+Green Fluorescent Protein (STM+GFP) reported in
Jeong~\emph{et al.}~\cite{Jeong2024TheZebrafish}, and Exosomes.
To estimate the diffusion coefficients of these, when and if not
available in the literature, we use the Stokes--Einstein relation:
\begin{equation}
    D = \frac{k_B T}{6\pi \mu R},
    \label{eq:D_stokes_einstein}
\end{equation}
which is an expression for the diffusion coefficient of a spherical
molecule suspended in a viscous fluid~\cite{Einstein1905UberTeilchen}.
In~\eqref{eq:D_stokes_einstein}, $k_B = \num{1.38e-23}$ J/K is the Boltzmann constant,
$T = 310$ K is the absolute temperature, $\mu$ is the dynamic viscosity of the
solvent (in this case the CSF), and $R$ is the radius of the molecule.
For the exosomes, we calculate $D$ using~\eqref{eq:D_stokes_einstein}
with $R=200$ nm. This radius covers the slowest diffusing exosomes,
since the smaller exosomes would have diffusion coefficients similar to the other molecules.
For STM+GFP, we estimated the diffusion coefficient by extrapolating
the experimentally observed diffusion coefficient of
GFP~\cite{Swaminathan1997PhotobleachingDiffusion, Potma2001ReducedCells},
assuming that if $D$ scales with $1/R$ according to~\eqref{eq:D_stokes_einstein},
$D$ would scale with the cube root of the molecule mass~\cite{Goodhill1997DiffusionGuidance}.
The resulting diffusion coefficients used in the simulations
are $D_1$ (Exosomes), $D_2$ (STM+GFP) and $D_3$ (Dendra2)~(\Cref{tab:diff_coeff_table}).

%Implicit in this approach is the assumption that the proteins all have the same density.

\begin{table}[h!]
    \centering
    \begin{tabular}{l|ccc}
        \toprule
        Molecule & Mass [kDa] & $D \ [\mathrm{m^2/s}]$ & Reference \\ 
        \midrule 
        Dendra2 Fluorescent Protein & 25.6~\cite{Gurskaya2006EngineeringLight}
        & $\num{1.15e-10}$  & \cite{GuraSadovsky2017MeasurementExpansion}\\
        GFP (saline aqueous solution) & 26.9~\cite{UniProt2024GFPUniProtKB}
        & $\num{8.70e-11}$ & \cite{Swaminathan1997PhotobleachingDiffusion, Potma2001ReducedCells} \\
        STM+GFP & 66.2~\cite{UniProt2024StmUniProtKB} + 26.9~\cite{UniProt2024GFPUniProtKB}
        & $\num{5.75e-11}$ & * \\
        Exosomes (radius of 200 nm) & -- & $\num{1.63e-12}$ & \textdagger \\
        \bottomrule
    \end{tabular}
    \caption{Molecular mass and diffusion coefficients of simulated solutes.
    STM and GFP denote Starmaker Protein and Green Fluorescent Protein, respectively.
    (*) The value was extrapolated based on the value of $D$ for
    GFP reported in the literature~\cite{Swaminathan1997PhotobleachingDiffusion, Potma2001ReducedCells},
    assuming that $D$ scales with the cube root of the molecule mass.
    (\textdagger) The value was estimated using the Stokes-Einstein relation~\eqref{eq:D_stokes_einstein}.}
    \label{tab:diff_coeff_table}
\end{table}

\subsection*{Quantities of interest}
We calculate the mean solute concentrations
\begin{equation}
    \cbar_i(t) = \frac{1}{|\Omega_i|}\int_{\Omega_i} c(\xx, t)\,\dx,
    \label{eq:c_mean_i}
\end{equation}
in the six regions of interest (ROIs) $\Omega_i$ ($i = 1, \dots, 6$)
as functions of time, where $|\Omega_i|$ is the volume of
the domain $\Omega_i$. The dynamics of ROI 1 follows directly
from~\eqref{eq:photoconversion_curve}. In addition,
we report time-to-threshold: the time $\hat{t}_i$ when
the mean concentration $\cbar_i$ first exceed a threshold
value $\hat{c}_i$. We use $\hat{c}_i=0.25$ for ROI 1--4 and
$\hat{c}_i=0.10$ for ROI 5--6.

To evaluate the relative importance of advection and
diffusion as transport mechanisms, we consider the Péclet number
\begin{equation*}
    \mathrm{Pe} = \frac{\mathrm{diffusion \ timescale}}{\mathrm{advection \ timescale}}
    = \frac{t_d}{t_a} = \frac{L_p^2/D}{L_p/U_p} = \frac{L_p U_p}{D}.
\end{equation*}
Thus, if $\mathrm{Pe} > 1$, transport is dominated by advection,
and if $\mathrm{Pe} < 1$ transport is dominated by diffusion.
Here, $L_p$ and $U_p$ are characteristic length and velocity scales,
respectively, and $D$ is the solute diffusion coefficient.
In calculating a global Péclet number, we use $L_p=600$ \textmu m, the approximate length of the
ventricles geometry along the rostrocaudal ($x$-) axis.
We use a mean velocity $\overline{U}$ as the characteristic velocity $U_p$.
The mean velocity $\overline{U}$ is calculated by first averaging the velocity in space:
\begin{equation*}
    \overline{u}^2 = \frac{1}{|\Omega|}\int_{\Omega}\uu\cdot\uu\dx,
\end{equation*}
and then averaging $\overline{u}$ in time over one cardiac cycle. To assess the assumption that the CSF flow is governed by the Stokes equations,
we also compute the Reynolds number
\begin{equation*}
    \mathrm{Re} = \frac{\mathrm{inertial \ forces}}{\mathrm{viscous \ forces}} = \frac{\rho U_r L_r}{\mu},
\end{equation*}
with $L_r$ and $U_r$ as length and velocity scales characteristic of the flow. 

%-----DISCRETIZATION AND MODEL IMPLEMENTATION-----%
\subsection*{Numerical approximation of the Stokes equations}

The Stokes equations~\eqref{eq:stokes_eqs} are discretized with a finite element method
in space, while in time, we employ a first-order implicit Euler discretization.
We set the timestep size $\Delta t = 0.02252$ s such that 20 timesteps represent one cardiac cycle.
For the spatial discretization, we use piecewise linear Brezzi-Douglas-Marini (BDM) elements for the velocity.
The BDM elements have degrees of freedom associated with integral moments of the
velocity component normal to the facets of elements~\cite{Brezzi1985TwoProblems},
thus we enforce the impermeability condition $\uu\cdot\nn=0$ strongly.
We impose the boundary conditions for both the tangential traction $\bsigpar$ and
the normal pressure $\Tilde{p}(t)$ weakly.
For the pressure, we use piecewise constant (zeroth order discontinuous Galerkin (DG)) elements.
The resulting total number of degrees of freedom at each time step is 943 820.
\fixme{The element combination $\mathrm{BDM}_1$--$\mathrm{DG}_{0}$ yields a non-conforming
discretization scheme for the Stokes equations~\cite{Stenberg1989SomeEquations}.
To ensure stability of the scheme, continuity in tangential velocities on interior facets
is weakly enforced with a penalty parameter $\gamma=10$~\cite{Hong2016AEquations}.}
Since the divergence of the velocity is a subspace of the pressure space for this scheme,
the divergence-free condition of the velocity is satisfied exactly on each
element~\cite{Boffi2008FiniteProblem}, ensuring mass conservation. 

\subsection*{Numerical approximation of the advection-diffusion equation}
The advection-diffusion equation~\eqref{eq:adv_diff_strong} is discretized
with a first-order implicit Euler method in time, and a first-order,
symmetric interior penalty discontinuous Galerkin (DG)
method in space~\cite{Arnold1982AnElements}.
To accurately approximate the solute concentrations even
in the presence of strong advection (high Péclet numbers),
we use an upwind scheme for the velocity when calculating the
advective flux~\cite{Patankar2018NumericalFlow}.
As for the flow equations, we use a timestep size of
$\Delta t = 0.02252$ s. We simulate transport for 1900 cardiac cycles,
with a final time of simulation $T = 856$ s.
A DG penalty parameter $\alpha = 25$ was chosen to ensure stability of the method. 

\subsection*{Solution strategy and software implementation}
Owing to the one-way coupling between the velocity field $\uu$
and the concentration $c$, the governing equations for the
CSF flow and the concentration can be solved sequentially when modeling solute transport.
\fixme{For the cilia-only/no-cardiac flow scenario, the velocity quickly reaches steady-state.}
In this case, we first solve the \fixme{steady-state} Stokes equations~\eqref{eq:stokes_eqs}
for $\uu$ and $p$, and then use the velocity field $\uu$ as input when solving
the advection-diffusion equation~\eqref{eq:adv_diff_strong}.
For the other flow scenarios, the Stokes equations are solved for one
period of the pulsatile cardiac motion. The solution of this one period
is then used periodically as input for the advection-diffusion equation.

The numerical solution schemes for the Stokes equations and the
advection-diffusion equation were implemented and solved numerically with DOLFINx,
the Python interface of the finite element software
FEniCSx~\cite{TheFEniCSProject2024FEniCSxDocumentation}, with
petsc4py~\cite{Dalcin2011ParallelPython} as linear algebra backend.
A direct solver using MUMPS~\cite{Amestoy2011Mumps} was used to solve the
linear systems resulting from the discretized Stokes equations.
For the discretized advection-diffusion equation, we used an iterative
linear solver based on GMRES~\cite{Saad1986GMRES:Systems} with block Jacobi
preconditioning~\cite{Jacobi1845UeberGleichungen}.
Numerical experiments verified the accuracy of the model implementation
(\Cref{sec:appB_verification}). The simulation code is openly
available~\cite{zenodo-link-to-all-code}. 

%%------PHYSICAL EXPERIMENTS------%%
\subsection*{Zebrafish maintenance and strains in photoconversion experiments}
The animal facilities for zebrafish~(\emph{danio rerio}) are approved by the
Norwegian Food Safety Authority (NFSA, Mattilsynet).
The zebrafish were maintained in accordance with the guidelines set by the
NFSA and the European Communities Council Directive.

The larval and adult zebrafish were raised under standard husbandry
conditions at 28.5°C in a Techniplast Zebtech Multilinking system.
The fish tanks were kept at constant pH 7.0 and 685 $\mu$S with a
14/10 hr light/dark cycle.
From fertilization to 3 dpf, larvae were maintained in egg
water (1.2 g marine salt and 0.1\% methylene blue in 20 L reverse osmosis (RO) water)
and subsequently transferred to artificial fish water (AFW)
(1.2 g marine salt in 20 L RO water). The zebrafish lines used in the experiments were
\emph{ccdc103(dnaaf19)\textsuperscript{tn222a}}
(\emph{schmalhans (smh)})~\cite{Jau-NianChen1997Left-rightZebrafish}
and \emph{Tg(gfap:Gal4FF)\textsuperscript{nw7Tg}}~\cite{DiazVerdugo2019Glia-neuronSeizures},
\emph{Tg(5xuas:Signal-Dendra2)\textsuperscript{nw21Tg}}.
Animals were in the pigmentless \emph{nacre\textsuperscript{b692}}
(\emph{mitfa\textsuperscript{-/-}})~\cite{JamesA.Lister1999NacreFate} background.

Wholemount~\emph{in vivo} live imaging and photoconversion
experiments were performed with zebrafish larvae at 2 days post-fertilization (dpf) stage obtained from inbreeding of heterozygous
\emph{smh\textsuperscript{+/-};Tg(gfap:Gal4FF);Tg(5xuas:Signal-Dendra2})
adult animals, as described in D'Gama~\emph{et al.}~\cite{DGama2024Cilia-mediatedBrain}.
Controls consisted of either wild-type (\emph{smh}\textsuperscript{+/+})
or \emph{smh}\textsuperscript{+/-} heterozygous zebrafish from the same breeding.
The mutants were identified based on their curved body~\cite{Jau-NianChen1997Left-rightZebrafish}.

\subsection*{Genotyping of adult zebrafish}
Genomic DNA (gDNA) was isolated from clipped fins of anesthetized adult
fish using 100 $\mu$L of PCR lysis buffer (containing 1M tris pH 7--9, 0.5 M EDTA,
tritonX-100, and Proteinase K 0.1 mg/ml) overnight at 50°C.
To stop the lysis reaction, the samples were heated to 95°C for 10 minutes
and then centrifuged at 13000 rpm for 2 minutes.
The supernatant containing gDNA was utilized for KASP assays-based analysis.
The gDNAs were diluted (1:2) with water, and 3 $\mu$L of the diluted gDNA was
used to perform the KASP assay following the guidelines of the manufacturer (LGC Biosearch Technologies™).
For each sample well, the master mix contained: 5 $\mu$L of master mix; 0.14
$\mu$L of assay mix; 1.86 $\mu$L of RO water.

\subsection*{Generation of \emph{Tg(5xuas:Signal-Dendra2)\textsuperscript{nw21Tg}} line}
\lyng{Needs a more understandable title.}
The open reading frame (ORF) of the neuropeptide y (npy) signal peptide
(\emph{npy}: ENSDARG00000036222, Q1LW93) was fused with the N-terminal
of zebrafish codon optimized Dendra2 DNA sequence.
The DNA sequence was synthesized with EcoR1 enzyme sites at the 5’ and
3’ ends (GenScript Biotech), and inserted into the
pT2MUASMCS vector~\cite{Asakawa2008GeneticZebrafish}
through restriction enzyme cloning (GenScript Biotech). 

To generate the transgenic line, the 5xuas:Signal-Dendra2 plasmid DNA was used.
A volume of 2 nl of a mixture of the plasmid DNA (60 pg) and tol2 mRNA (10 pg)
was microinjected into one-cell stage embryos,
as described in Jeong~\emph{et al.}~\cite{Jeong2024TheZebrafish}. The injected
embryos were raised to adulthood (F0).
Germline-transmitted founder zebrafish were identified by breeding with
multiple Gal4 transgenic lines. Stable F1 embryos expressing the Gal4-driven
Dendra2 signals were screened and raised to adult zebrafish.

\subsection*{Wholemount zebrafish \emph{in vivo} live time-lapse imaging and photoconversion of Dendra2}
Two dpf larval zebrafish were anesthetized in 0.013\% MS222
(Ethyl 3-aminobenzoate methanesulfonate, Sigma) in AFW and
mounted laterally in 1.5\% low melting agarose in a Flurodish (VWR, FD35PDL-100),
to obtain a lateral view of the brain ventricles. After the
fish were positioned in the agarose, the dish stood for 5 min to allow the agarose to solidify.
After solidification, AFW containing 0.013\% MS222 was added,
and then the fish were transferred to the confocal microscope (LSM880 Examiner, Zeiss)
and imaged using a 20x water-immersion objective (Zeiss, NA 1.0, Plan-Apochromat)
at room temperature.

The time-lapse images were acquired in a single plane, covering the
telencephalic, diencephalic and rhombencephalic ventricles simultaneously,
with a frequency of 0.37--0.38 Hz (2.60--2.68 sec / frame). The size
of the images was $1024\times400$ pixels. In total, 300 images were acquired per fish.
The first 10 images were scanned without photoconversion to obtain a
baseline value of fluorescence intensity, and the next 290 images were
scanned while performing photoconversion. 

For photoconversion, ‘Bleaching’ and ‘Region’ in the ZEN software
were used. A 405 nm laser was focused into a circular area (diameter: 16 $\mu$m, scan 0.77 $\mu$sec/pixel)
of the dorsoposterior diencephalic ventricle with 100\% laser power.
The laser was illuminated in the area repeatedly between each scanning.
After imaging, the fish health was checked and only data from healthy fish were analyzed.
Data were collected and analyzed from three separate experiments,
with a total of 8 controls and 6 mutants.

\subsection*{Photoconversion data analysis}
All data analysis was performed in MATLAB and the codes are available in
the repository of this article: \url{repo-link}.

Following import of the image data, the photoconverted channel
was aligned to correct for drift in $x,y$ directions using previously
developed codes~\cite{Reiten2017Motile-Cilia-MediatedComputations, Ringers2023NovelEpithelia}.
Only stable recordings were further analyzed. 
To increase the signal to noise ratio, the images were downsampled by
a block average using a resampling factor of 6 (using the MATLAB function blockproc).
The change of fluorescence $($$(F-F_0)/F_0$, hereafter referred to as
$\Delta F$$)$ was calculated for each pixel at each time point of the data.
Here, the value $F_0$ is the average fluorescence intensity for the
baseline (the first 10 time-lapse images) acquired before photoconversion. 

To identify the time needed to pass a certain threshold value,
$\Delta F$ values were first smoothened.
The first timepoint when values surpassed the threshold was reported
per pixel (in the downsampled images).
A mask for the image was generated using the green (not photoconverted)
channel and calculated on the block-averaged data based on the intensity
being higher than 1.5x median intensity of all the frames.

To obtain $\Delta F$ values for a region of interest (ROI), ROIs were
first drawn on the aligned and block-averaged time series.
Six ROIs were drawn manually: ROI 1 around the location of photoconversion,
and ROIs 2--6 in different regions of the ventricular system.
The pixel values of the fluorescence intensity within one ROI were
averaged for each time point, and the relative change $\Delta F$ calculated with this averaged value.
To normalize the data with respect to photoconversion efficiency, the
$\Delta F$ curves for all ROIs were divided by the $\Delta F$ values
obtained for ROI 1 at the end of the experiment,
so that $\Delta F$ values for the photoconverted site ROI 1 ranged from
zero to one. Finally, we averaged the $\Delta F$ data from controls and from mutants.
We report these mean values together with a one-standard deviation error band.

\subsection*{Intraventricular injection of microbeads and particle tracking}
To recover the flow fields shown in~\Cref{fig:figure1}b, c, we carried out
particle tracking in a larval zebrafish injected with fluorescent beads.
As described in Olstad~\emph{et al.}~\cite{Olstad2019CiliaryDevelopment},
anesthetized 2 dpf zebrafish were injected with 1 nl of a mixture containing 0.1\% w/v fluorescent beads
(SPHERO Fluorescent Yellow Particles 1\% w/v, F = 0.16 mm) diluted
in 7.5 mg/ml 70 kDa rhodamine B isothiocyanate-dextran (RITCdextran; Sigma-Aldrich, R9379)
dissolved in artificial CSF. Artificial CSF composition was as follows:
124 mM NaCl, 22 mM D-(+)-Glucose, 2.0 mM KCl, 1.6 mM MgSO\textsubscript{4} $\cdot$ 7 H\textsubscript{2}O,
1.3 mM KH\textsubscript{2}PO\textsubscript{4}, 24 mM NaHCO\textsubscript{3},
2.0 mM CaCl\textsubscript{2} $\cdot$ 2 H\textsubscript{2}O.
The needles used for the injections were pulled with a
Sutter Instrument Co. Model P-2000 from thin-walled glass
capillaries (1.00 mm; VWR) and cut open with a forceps.
A volume of 1 nl of the solution was injected with a pressure injector
(Eppendorf Femtojet 4i) in the rostral rhombencephalic ventricle.
The pressure and time were calibrated for each needle using a 0.01 mm calibration slide.

Following injection, the zebrafish was directly placed under the confocal
microscope and 1200 images were acquired at a frequency of 13.1 Hz.
A single optical section was obtained. Particle tracking was done using
TrackMate~\cite{Tinevez2017TrackMate:Tracking} in Fiji/ImageJ~\cite{Schindelin2012Fiji:Analysis}
and plotted using MATLAB, as described in Jeong~\emph{et al.}~\cite{Jeong2022MeasurementTelencephalon}.
The parameters for TrackMate were as follows.
The DoG detector was used with the settings: threshold: 2.0, with
median filtering; radius: 2.0, with subpixel localization.
Next, the Simple LAP tracker was used with settings: max frame gap: 2;
alternative linking cost factor: 1.05; linking max distance: 5.0; gap closing max distance: 2.0;
splitting max distance: 15.0; allow gap closing: true; allow track splitting:
false; allow track merging: false; merging max distance: 15.0; cutoff percentile: 0.9.
Only the particle tracks with at least 30 data points, while simultaneously
covering a distance of at least 8.8 $\mu$m, were plotted in~\Cref{fig:figure1}b, c.

%%--------------RESULTS--------------%%
\section*{Results}

\subsection*{Motile cilia induce flow compartmentalization within the brain ventricles}

We seek to model the compartimentalized and pulsatile CSF flow observed \emph{in vivo}~(\Cref{fig:figure1}b--c).
To characterize CSF flow patterns predicted by
computational fluid dynamics within the brain ventricles, we begin by
considering the simulated CSF velocity and pressure when only considering
cilia motion, disregarding cardiac pressure pulsations. The
bidirectional traction induced by the motile cilia
inherently leads to a partial compartmentalization of the ventricular
system, with large-scale vortex structures in the anterior and middle
ventricles~(\Cref{fig:figure1}e). The highest velocities ($26.5$ \textmu m/s)
occur in the vicinity of the dorsal cilia in the middle ventricle. 

To examine the effect of cardiac pressure pulsations,
we next consider a no-cilia model variation. In this scenario, the simulated CSF
flow is pulsatile as a result of the sinusoidal pressure gradient (\Cref{fig:figure1}g),
with a peak velocity magnitude of $4.8$ \textmu m/s. At each time instant, the flow
is unidirectional, with CSF initially flowing rostrocaudally (positive $x-$axis) before changing
direction to caudorostral in the middle of the cardiac cycle.

We now consider the baseline model, simulating flow driven by both cilia
and cardiac pulsations. We observe \fixme{persistent} flow features,
with vortex-structures in the anterior and middle ventricles
and directional flow in the posterior ventricle (\Cref{fig:figure1}e,
\fixme{Supplementary video 1}). These characteristic patterns suggest
that the flow in the anterior and middle ventricles are dominated by
the cilia contribution, while the cardiac pressure pulsations dominate
within the posterior ventricle. Over one cardiac cycle, the CSF flow
speed reaches up to 27.7 \textmu m/s, dorsally in the posterior part
of the middle ventricle. This peak is attained at the time instant
coinciding with the time of peak caudal cardiac pulsatile flow, as a
result of both the traction induced by the cilia and the pulsatile
pressure-driven flow being aligned with the rostrocaudal 
axis. Using the observed maximum velocity magnitude and choosing the height of
the middle ventricle ($110$ \textmu m) as a characteristic length
scale, yields a Reynolds number $\mathrm{Re} \approx 0.004$,
justifying the assumption of Stokes flow. In summary, the computational
CSF flow model reproduces the flow features observed \emph{in vivo}.

\subsection*{Ventricular solute transport is advection-dominated for larger molecules}
We next ask: what is the impact of these CSF flow patterns and
magnitudes on solute transport generally within the brain ventricles?
To address this question, we simulate the distribution and evolution
of different solutes subject to photoconversion within a region of
interest in the dorsal region of the middle ventricle (ROI 1)
(\Cref{fig:fig2}\fixme{e/x}). In particular, we consider the spread of
a solute concentration with a given molecular diffusion coefficient,
corresponding to Exosomes ($D_1 = 1.63 \times 10^{-12}$ m$^2$/s),
STM+GFP ($D_2 = 5.75 \times 10^{-11}$ m$^2$/s), and Dendra2 ($D_3 =
1.15 \times 10^{-10}$ m$^2$/s). As expected, transport occurs more
rapidly with increasing diffusion coefficient. Interestingly, the
distribution patterns differ between the \fixme{different-characteristics}
solutes. For the smallest diffusion coefficient, representing
Exosomes, the advection induced by the CSF flow dominates diffusion
(\Cref{fig:fig2}a). The vortex structures of the velocity field
carry over into the concentration field, indicating that transport
along the streamlines is more rapid than the diffusion across the
streamlines. The balance between diffusion and advection shifts with
increasing diffusion coefficients, the CSF flow structures become more
obfuscated, and the solute spreads more uniformly throughout the
ventricular geometry (\Cref{fig:fig2}\fixme{x-y}). Notably, the
smaller solutes with higher diffusion coefficients spread more quickly
to regions distal to the photoconversion site (\Cref{fig:fig2}\fixme{z}).
\begin{figure}%[H]
    \centering
    \includegraphics[width=0.95\textwidth]{graphics/figure2_simulation_results_different_coeffs.png}
    \caption{Simulated distribution and evolution after photoconversion of
    Dendra ($D_1$), STM+GFP ($D_2$) and Exosomes ($D_3$).
    \mer{MER: I think you should show on the same color scale here in a)
    (the range of values lets you and then it is possible to compare).
    Split a into a, b, c. Show e) first (so that e) becomes a)}.
    \textbf{a.} Slices in the $xz$-plane ($y=0.140$ mm,
    the center of the geometry) showing the concentration field simulated
    with diffusion coefficients $D_1$, $D_2$ and $D_3$ for three
    time instants $t=45.0$ s, $t=112.6$ s, $t=450.5$ s.
    Note that the colorbars indicating the magnitude of $c$ have different limits for each column.
    \textbf{b--d.} Mean concentration $\cbar$ in each region of interest (ROI)
    as function of time for diffusion coefficient $D_1$, $D_2$ and $D_3$, respectively.
    \textbf{e.} Geometry-centered clip of the ventricles mesh showing the locations of the ROIs.
    \textbf{f.} The time $\hat{t}$ before the mean concentration $\cbar$ exceeds a
    threshold value of $0.25$ (ROIs 2-4) or $0.10$ (ROIs 5, 6) in each simulation setup.
    Note that there is no bar for $D_1$ in ROI 6, because the threshold value was never exceeded.}
    \label{fig:fig2}
\end{figure}

To quantify the transport of solute, we plot the profile of the mean
concentrations $\cbar_i(t)$ as functions of time
(\cref{fig:figure2_compare_diffusion_coefficients}\fixme{b--d}) and
the times $\hat{t}$ when the mean concentrations $\cbar_i$ first exceed
a threshold value $\hat{c}$ (\cref{fig:figure2_compare_diffusion_coefficients}\fixme{f})
in 6 regions of interest (ROIs) (\cref{fig:figure2_compare_diffusion_coefficients}\fixme{e}).
The dynamics at the site of photoconversion (ROI 1)
is logarithmic and follows from imposing \eqref{eq:photoconversion_curve}.
For ROIs 2--6, we observe faster kinetics with higher diffusion coefficients
(\Cref{fig:figure2_compare_diffusion_coefficients}\fixme{b--d}),
which results in smaller time to threshold
(\Cref{fig:figure2_compare_diffusion_coefficients}\fixme{f}).
For all three diffusion coefficients, the dynamics of ROI 5
is significantly delayed compared to ROI 2--4,
and ROI 6 is the slowest. Notably, for $D_1$ the mean concentration
in ROI 6 is close to zero at the end of simulation.
To quantify the balance of advection versus diffusion as transport mehcanisms,
we calculate the Péclet numbers $\mathrm{Pe}_1=519$,
$\mathrm{Pe}_2=14.7$ and $\mathrm{Pe}_3=7.4$ for $D_1$, $D_2$ and $D_3$, respectively.
Since $\mathrm{Pe}>1$, this indicates that advection is the
dominant transport mechanism on a \fixme{global} scale.
 
\subsection*{Validation of the simulations with experimental data}
To validate our computational model, we performed photoconversion experiments
in transgenic zebrafish larvae expressing a secreted Dendra2 in their brain ventricles.
Our photoconversion protocol consisted of the acquisition of a baseline,
before localized exposure to a UV laser (at the ROI 1 location)
that converts Dendra2 from a green to a red emitting fluorescent protein. 
To quantify the transport of photoconverted proteins, we measured and
calculated the change in fluorescence intensity $\Delta F = (F(t)-F_0)/F_0$
over time (\Cref{fig:figure3_compare_exp_and_sim_control}a,
$F_0$ being the average fluorescence at baseline). 

We obtained very similar kinetics curves (\Cref{fig:figure3_compare_exp_and_sim_control}a, b)
and time-to-threshold values (\Cref{fig:figure3_compare_exp_and_sim_control}c, d) when comparing our
experiments (\Cref{fig:figure3_compare_exp_and_sim_control}a, c) with
simulation (\Cref{fig:figure3_compare_exp_and_sim_control}b, d).
Importantly, since the initial condition of the simulation is a mean concentration of zero,
the mean concentration $\cbar$ can be interpreted as the change in mean concentration and
is thus a similar metric to $\Delta F$.
The final values of $\cbar$ were 0.76, 0.72, 0.82, 0.56 and 0.35 respectively,
compared to 0.63, 0.71, 0.67, 0.29 and 0.18 for $\Delta F$.
Notwithstanding similarities, we observed small differences between the experiments and simulation.
For instance, transport dynamics in ROIs 2--6 were faster in the simulation and reached higher final values.
Moreover, the $\cbar$ profiles of ROIs 3--4 were opposite order in
the simulations as compared to the experiments.
Despite observing small differences, the \emph{in vivo} experiments and
simulations were similar enough to convincingly validate the model.
\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{graphics/figure3_compare_experiments_and_simulations.png}
    \caption{\textbf{a.} Relative change in fluorescence intensity $\Delta F$ over time
    in each region of interest (ROI)
    observed during experiments of photoconversion of Dendra2 proteins in ROI 1.
    \textbf{b.} Mean concentration $\cbar$ as function of time in each ROI in a
    simulation of Dendra2 protein transport with
    the diffusion coefficient $D_3$.
    \textbf{c.} Time to $\Delta F$ reaches a threshold value of 0.25 in
    one representative zebrafish larva.
    Time to $\cbar$ reaches a threshold value of 0.25 for
    a transport simulation with diffusion coefficient $D_3$.
    We have excluded the posterior-most regions of the geometries,
    because the values of $\Delta F$ and $\cbar$ never exceeded the threshold
    value of 0.25 in these regions.}
    \label{fig:figure3_compare_exp_and_sim_control}
\end{figure}
% Final Delta F values
%0.6322650156762044 ROI 2
%0.7059449124175479 ROI 3
%0.6694500367160006 ROI 4
%0.28986700094045437 ROI 5
%0.18155055624229066 ROI 6
% Final mean c values ROIs 2-6: 0.7637806  0.72025859 0.81903216 0.56204164 0.34926411

\subsection*{Absence of ciliary motion affects local protein distribution}
Since CSF flow is highly disturbed in the absence of motile cilia, we
want to assess the impact of cilia motion on solute distribution. To this end,
we measured the movement of photoconverted Dendra2 signal
in control and motile-cilia mutant zebrafish (\emph{schmalhans} (\emph{smh})).
By plotting the means of $\Delta F$ (\Cref{fig:figure4_compare_control_mutant}a)
and the time to threshold
across the control and mutant cohorts (\Cref{fig:figure4_compare_control_mutant}b),
we identified a significant
difference in Dendra2 distribution in ROIs 2 and 4, but not in ROIs 3, 5, and 6. In ROIs 2 and 4,
which measure Dendra2 distribution downstream of the photoconversion site,
we observed a slower increase of fluorescence intensity,
resulting in $\sim X\%$ (ROI 2) and $\sim 45\%$ (ROI 4)
higher mean times-to-threshold in the \emph{smh}
mutants compared to controls (\Cref{fig:figure4_compare_control_mutant}b). 

Next, to identify whether our model would recapitulate these findings,
we simulated protein transport by using the only-cardiac/no-cilia flow model 
and compared it with the baseline model
(both cilia and cardiac motion) (\Cref{fig:figure4_compare_control_mutant}c).
Similar to our experiments, we observed slower dynamics in simulations for ROIs 2 and 4.
Interestingly, the difference was even greater for the simulations,
with the time-to-threshold in ROI 2 being 90\% higher in absence of cilia-mediated flow.
Moreover, ROI 5 exhibits very different $\cbar$ profiles for the two
configurations in our computational model,
but not in our experimental data.
Altogether, our findings identify that diffusion plays an important
role for solute movement to regions upstream in the cilia flow fields or to distant ventricles.
\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{graphics/figure4_compare_control_mutant.png}
    \caption{\textbf{a.} Relative change in fluorescence intensity $F$ over
    time in each region of interest (ROI) for control (black lines) and \emph{smh} mutant (red lines) fish. 
    \textbf{b.} Times when $\Delta F$ first exceeds a threshold value,
    equal to 0.25 (ROIs 2--4) or 0.10 (ROIs 5, 6), for control (gray dots) and
    \emph{smh} mutant (red dots) fish.
    The thick black lines denote mean values of the datasets, and $p$-values
    were calculated using a Mann-Whitney U test with a confidence level of 95\%.
    \textbf{c.} Mean concentration $\cbar$ as function of time in each ROI,
    simulated using the baseline model (black lines) and the cardiac-only/no-cilia model (red lines).
    The horizontal gray lines mark the threshold value $\hat{c}$, and the time
    $\hat{t}$ when the threshold is first exceeded is reported in the plot legends.
    Note that in both panels \textbf{a} and \textbf{c}, the vertical axes have
    different limits in each plot.}
    \label{fig:figure4_compare_control_mutant}
\end{figure}

\subsection*{Loss of cilia motility in specific subpopulation slows down transport}
We previously identified three distinct ciliated cell lineages in the
zebrafish larval brain~\cite{Olstad2019CiliaryDevelopment, DGama2025MotileBrain}.
To identify the role of these lineages on CSF transport, we modified flow boundary conditions
in the three different ciliated regions
(\Cref{fig:figure5_compare_cilia_modifications}a, magenta markers)
and simulated CSF flow using the Stokes equations.
We identified that the velocities were most strongly
affected when removing cilia in the dorsal diencephalic cilia,
as the maximum velocity dropped from 27.7 $\mu$m/s to 8.0 $\mu$m/s.
In contrast, removing telencephalic and ventral diencephalic had minimal
impact and resulted in maximum velocities of 27.7 $\mu$m/s or 27.2 $\mu$m/s, respectively.

With the resulting velocity fields, we simulated transport of Dendra2.
Changes in the transport dynamics were most prominent when removing the dorsal cilia
in the diencephalic ventricle, which produces the strongest flow at this developmental stage.
All ROIs (except ROI 1, where we locally released solutes) exhibit significant changes
in the dynamics of $\cbar$ (\Cref{fig:figure5_compare_cilia_modifications}b).
Compared to the baseline model results, we observed a slower increase and lower final concentration
in the middle and anterior ventricles (ROIs 2--5),
and a faster transport to the posterior ventricle (ROI 6).
In contrast, removal of telencephalic cilia did not alter the evolution of the
mean concentration $\cbar$ with time (\Cref{fig:figure5_compare_cilia_modifications}b).
Finally, without ventral cilia in the diencephalic ventricle,
more solute has spread towards the anterior ventricle (ROI 5) and
less solute has spread towards the posterior ventricle (ROI 6).
\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{graphics/figure5_compare_cilia_modifications.png}
    \caption{\textbf{a.} The three separate ciliated regions where loss of cilia motility is simulated
    by imposing a zero tangential traction boundary condition in the respective region.
    \textbf{b.} Mean concentration $\cbar(t)$ for the six regions of interest in the three 
    loss-of-cilia-motility scenarios considered. The region named in the legend is the region
    where we modify the cilia tangential traction boundary condition to a free-slip condition
    in the CSF flow model.}
    \label{fig:figure5_compare_cilia_modifications}
\end{figure}

\subsection*{Modification of ventricles geometry impacts protein distribution}
To assess the impact of ventricular morphologies on solute distribution,
we used four alternative brain ventricles meshes:
one where the connection between the anterior and middle ventricles is constricted;
one where the connection between the middle and posterior ventricles is constricted;
one where the middle ventricle is shrunk in the lateral direction;
and the combination of the above three modifications (\Cref{fig:figure6_sim_results_compare_geometries}a).
We simulate CSF flow in the modified geometries with the baseline flow model
and use the resulting velocity field to simulate transport of Dendra2 (\Cref{fig:figure6_sim_results_compare_geometries}b).

Constricting the connection between the anterior and middle ventricle leads to
slower transport to the anterior ventricle (ROI 5),
and a slightly higher mean concentration in ROIs 2--4 and 6, compared to the original geometry.
On the other hand, constricting the connection between the middle and the posterior ventricle
leads to less transport into the posterior ventricle (ROI 6),
and slightly more transport towards the anterior ventricle (ROI 5).
Faster transport to the anterior ventricle (ROI 5) and slower transport to the
posterior ventricle (ROI 6) is also the case for the geometry with a shrunk middle ventricle.
Additionally, for this configuration the transport upstream the flow fields towards the
ventral-posterior region of the middle ventricle (ROI 3) is slowed down.
For the mesh where all three ventricles are modified,
we observe higher retention of solute in the middle ventricle (ROIs 2--4),
while both the anterior and the posterior ventricles (ROIs 5 and 6) have lower concentration.
Altogether our results identify that the geometry and width of the ventricular ducts matter
for solute transport and may explain why we observed larger differences
in the cilia-deficient model than in the experimental data in
\Cref{fig:figure4_compare_control_mutant}, which have different duct diameters.

\lyng{HH: Quantify differences in terms of percentage changes in the
final mean concentration values, maybe with a bar plot?}
\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{graphics/figure6_compare_modified_geometries.png}
    \caption{Caption. \mk{MK: How do the different geoemtries compare in terms of volume,
    areas where cilia are active?} \lyng{HH: Good question, will add this information.}}
    \label{fig:figure6_sim_results_compare_geometries}
\end{figure}

%%--------------------------------------%%
%%--------------------------------------%%
%%--------------DISCUSSION--------------%%
%%--------------------------------------%%
%%--------------------------------------%%

\section*{Discussion}
We have presented a computational model of CSF flow and solute transport within the brain ventricles.
In this study, we used the larval zebrafish brain ventricles since its geometry,
flow components and kinetics are well described, thereby allowing us to validate
our simulation experimentally. Nevertheless, our simulations can be implemented in
other systems in the future when their flow parameters are comprehensively resolved.  
By imposing tangential traction and normal pressure on the CSF flow, we achieved to reproduce
flow fields with similar features as observed experimentally, consisting of vortex-structured flow,
mediated by cilia, and pulsatile flow, mediated by the cardiac cycle.

Using the simulated CSF flow, we modeled solute transport by an advection-diffusion equation.
These results compared well with experimental data performed \emph{in vivo} in zebrafish.
Next, by using both a model and zebrafish mutants with paralyzed cilia,
we identified that even though advection by cilia is dominant,
diffusion plays a significant role  in protein transport. Finally, we report how local
flow patterns and ventricular geometries affect solute distribution.
Altogether our results provide not only new findings on the role of
diffusion and advection but also offer a new framework to study CSF transport in complex geometries.

\subsection*{The roles of advection and diffusion in intraventricular solute transport}
To investigate the roles of advection and diffusion in solute transport, we quantified transport
based on changes in fluorescence intensity $\Delta F$ (experiments) and
mean concentration $\cbar$ (simulations) as functions of time.
Even though we identified very similar dynamics in simulations and
\emph{in vivo} experiments, the dynamics of the simulations were generally
more rapid and resulted in higher total amounts.
Differences between experiments and simulations may relate to
ventricular size, as we showed that modifying the duct volumes impacts
solute distribution and the geometry used for the model was larger than in experiments.
Alternatively, it may relate to the way we defined the photoconversion site (region of interest 1)
in the simulations, since the volume of the photoconversion region determines
the total amount of solute, and thereby the rate of transport. Another source
of discrepancy may relate to the permeability of the ventricular wall,
which we have ignored in our simulation. Reports have shown that the
neuroepithelium lining the walls of brain ventricles in 24h old zebrafish
is permeable to dyes smaller than 70 kDa~\cite{Chang2012AnNeuroepithelium}.
Since Dendra2 is around 26 kDa in size~\cite{Gurskaya2006EngineeringLight},
and thus under this reported limit, one could assume that if Dendra2
leaks out of the brain ventricles, the dynamics become slower.

In general, the mean concentration dynamics of the simulations was 
more rapid than the fluorescence intensity dynamics of the photoconversion experiments.
This is strongly related to our definition of the photoconversion site
(ROI 1) in the simulations, and whether this corresponds well to the
region used in experiments, because the volume of the photoconversion
region used in the simulations determines the total amount of solute,
and thus impacts the rate of transport. The alignment of the computational
and experimental region of interest definition is hard to quantify.
Additionally, a source of discrepancy in the results is the fact that the
brain ventricles might be permeable, such that some Dendra2 diffuses out
of the brain ventricles in the experiments. An assay has shown that the neuroepithelium
lining the walls of brain ventricles in embryonic zebrafish 24 hours post-fertilization
is permeable to dyes smaller than 70 kDa~\cite{Chang2012AnNeuroepithelium}.
Dendra2 are almost 26 kDa in size~\cite{Gurskaya2006EngineeringLight},
and are thus under this reported limit. If such permeability also is the case for 2 days
post-fertilization zebrafish embryo ventricles, this would lead to Dendra2 leaking
out of the brain ventricles in the experiments, making the dynamics appear slower. 

Nevertheless, our results convincingly show that intraventricular transport of proteins
depends on a balance of advection and diffusion. While advection is the strongest transport
mechanism on a global scale for Dendra2 (global Péclet number 7.4), diffusion is balancing out
advection especially for shorter distances and for regions devoid of motile cilia activity.
Indeed, we identified that the diffusion coefficient had a greater impact on transport to
regions located upstream in the CSF flow field (ROI 4 \fixme{@Nathalie: should this have been 3 instead of 4?}) or in different ventricles (ROIs 5--6),
than the regions located downstream of cilia-mediated flow (ROI 2).

In contrast, we identified that transport of larger molecules
(thus with smaller diffusion coefficients),
such as 200 nm exosomes, are almost entirely dependent on advection.
Indeed, there was essentially zero solute in the posterior ventricle at the
final simulation time (around 850 s), where transport is largely driven by diffusion.
For $D_1$, an estimate of the diffusion timescale $t_d=L^2/D_1$ over a distance
$L=200$ $\mu$m from the middle to the posterior ventricle is approximately 24500 s.
Advective transport by the mean velocity $\overline{U}=1.41 \ \mu$m/s over the same distance would happen
on a timescale of the order of $t_a=L/\overline{U}=141$ s. Comparing this with the minuscule values of
$\cbar_6$ after simulating transport with $D_1$ for 850 s, one can conclude that if advection
played a role in transporting solute from the photoconversion site in the middle ventricle
to the posterior ventricle, values of $\cbar_6$ would be higher than those observed.

\subsection*{Impact of cilia properties and ventricular geometry}
To gain further insights into how CSF distribution depends on cilia properties and geometries,
we manipulated certain parameters in our models. First, we removed cilia in specific regions of
our model (\Cref{fig:figure5_compare_cilia_modifications}). We identified that the cilia in the
dorsal middle ventricle contribute the most to solute distribution, which aligns well with the fact
that these cilia generate the strongest flow field \emph{in vivo} and in our model.
In contrast, removing the ventral cilia in the middle ventricle did not greatly impact flow dynamics.
However, it lead to reduced transport towards the posterior ventricle,
and increased transport towards the anterior ventricle, as the ventral cilia help induce
flow towards the posterior ventricles. Removing the cilia in the anterior ventricle did not affect the transport.

Next, we modified the ventricular geometry and identified varying impact on transport.
Notably, shrinking ducts reduced movement between the ventricles, while shrinking the middle ventricle
reduced diffusive transport upstream of the flow field (ROI 4).
We previously identified large inter-individual variation in ventricular size and
CSF velocities between larva at given developmental stages and throughout development~\fixme{@Nathalie: Add some citation here? Olstad/D'Gama}.
Our results suggest that these two parameters may well co-vary and explain the variation.

Altogether, our model is flexible and thereby allows \emph{in silico} testing of
further hypotheses on the contribution of ciliated lineages and geometries.
We trust that it will be useful as a basis for future analysis of CSF dynamics in more complex geometries.

\lyng{HH: Compare numerical results with transport results in Figure 7 in \cite{Olstad2019CiliaryDevelopment}.}

\lyng{HH: Discuss cilia serving as an "enhanced mixing factor" on small scales,
refer to this~\cite{Siyahhan2014FlowVentricles} and this~\cite{Yoshida2022EffectVentricles}
paper that studied this in human brain ventricles. Include a discussion of what happened when these papers removed cilia.}

\subsection*{Computational model assumptions and boundary conditions}
To model CSF, we employed slip boundary conditions at the ventricular walls,
contrasting the classical choice of a noslip boundary condition.
The use of slip boundary conditions is motivated by experimental observations of the
flow field in zebrafish embryo brain ventricles~\cite{Olstad2019CiliaryDevelopment}.
It has been observed, both previously and in this work, that the fluid velocity increases towards the wall~(\Cref{fig:figure1}c).
Using slip boundary conditions allows for this behavior.
A couple of previous continuum model simulation studies of cilia-mediated flow~\cite{Siyahhan2014FlowVentricles, Thouvenin2020OriginCanal}
have modeled cilia as body forces. We chose to model the cilia as a force exerted tangentially on the boundary.
First, we deem this a good representation of the real nature of cilia,
since they are in reality microscale structures confined to the ventricular walls that
generate flow through collective motion. Second, it avoids the need of excessive mesh
refinement near the ciliated boundaries in the case that a
body force were to be applied in a $\sim$ 5 $\mu$m~\cite{Salman2022ComputationalEmbryo} deep cilia layer.

In our CSF flow model, the ciliated boundaries are homogeneous,
whereas in reality, cilia populate the ventricle walls in a
heterogeneous manner~\cite{Olstad2019CiliaryDevelopment}. Heterogeneous spatial
arrangement of cilia, both in location and beating direction,
has been shown to enhance particle clearance in airway cilia arrays
in mice~\cite{Ramirez-SanJuan2020Multi-scaleArrays}. Incorporating spatial arrangement heterogeneity
in our model could possibly result in different transport timescales,
but we would not expect notable changes to the general trends of the macroscopic
transport that we study here, since the effects of introducing heterogeneous spacing
of cilia would alter the magnitude of cilia forces, but not their direction.
On the other hand, introducing more heterogeneous beating directionality could
alter the transport timescales as well as the directionality. In our model, all of the applied cilia
tangential traction is aligned with the rostrocaudal axis, and more heterogeneity
in this alignment to reflect heterogeneity in the beating directionality of
cilia would presumably affect both magnitude and directionality of advective transport.
Previous studies have demonstrated that changes to beating directionality affects transport,
and that specific coordination of the ciliary beating is important in the brain ventricles for
flow and signaling processes~\cite{Eichele2020Cilia-drivenVentricle, Olstad2019CiliaryDevelopment,
Faubel2016Cilia-basedVentricles, Guirao2010CouplingCilia, Afzelius2004Cilia-relatedDiseases},
but also in other phenomena such as maintaining healthy functioning of mucous transport in
airways~\cite{Rayner1996CiliarySyndrome, Schneiter2021Multi-scaleFunction,
Bustamante-Marin2019LackClearance, Tsukita2012CoordinatedFeet} and in nodal flow which establishes
left-right symmetry in mammals~\cite{Yoshiba2014RolesSymmetry, Hirokawa2006NodalAsymmetry,
Sawamoto2006NewBrain}. Although these studies were mostly on spatial scales resolving individual cilia,
we could envision the same effects in our continuum model,
similar to the results of previous numerical studies~\cite{Ramirez-SanJuan2020Multi-scaleArrays,
Thouvenin2020OriginCanal, Yoshida2022EffectVentricles}. 

We used an advection-diffusion equation to simulate protein transport in the brain ventricles.
Except for where we imposed a normal pressure boundary condition on the CSF flow,
we imposed a no-flux condition on the concentration. Imposing a pressure-driven
cardiac-related flow on certain parts of the CSF flow boundary leads to in- and outflow,
and the advective transport then requires handling of in- and outflux boundary conditions.
Reflecting our assumption that the ventricular system is closed, we set boundary conditions such
that the total mass is conserved in the limit that the timestep size is zero.
We handle the outflux by leaving the advective flux term as an unknown in the weak formulation
of the advection-diffusion equation. This has been referred to with different names in previous works,
names such as the ``free boundary condition"~\cite{Papanastasiou1992ACondition},
the ``no boundary condition"~\cite{Griffiths1997TheCondition}, and the 
``consistent flux boundary condition"~\cite{Lynch2020NumericalHemodynamics}. 
In leaving the advective flux term in the weak form we are lacking a boundary condition,
so the approach seems problematic in the continuous sense of the problem.
However, the discrete problem is well-posed, because the approach represents
constraining higher-order derivatives of the concentration~\cite{Griffiths1997TheCondition}. 

On the inflow boundary, we applied an approximate periodic boundary condition by setting
the inflow boundary flux at timestep $n$ equal to the outflow boundary flux at the previous
timestep $n-1$. This is expected to be of low-order accuracy, and for high Péclet numbers,
the boundary condition affects the concentration field in the vicinity of the in- and outflow boundaries
due to large advective fluxes. For the simulation scenarios we considered, both in terms of model parameters
and time span, the photoconversion region where solute originates is far enough away from the in- and outflow
boundaries for the flux boundary condition to not significantly impact the results. Nevertheless,
we deliberately excluded the space proximal to in- and outflow boundaries in
the anterior and posterior regions of interest (5 and 6) to limit adverse effects. 

A possible improvement to the in- and outflow boundary conditions we employed would be
to consider a 3D-0D coupling of the in- and outflow boundaries using ordinary differential equations,
similar to what has been done in cardiac modeling~\cite{Brown2024AMechanics, Augustin2021ACirculation}.
However, if the ventricular system in reality is a closed system,
one should rather seek out a way of modeling the CSF flow without open boundaries to avoid
handling in- and outflow boundary conditions. A more appropriate way of handling the beating
cardiac motion could be to impose a ventricular wall deformation, assumed to result from pulsatile
motion of the heart and/or cardiac vessels surrounding the ventricles. This wall deformation would
induce CSF flow while simultaneously permitting enforcement of the impermeability condition
$\uu\cdot\nn=0$ on the ventricular walls. Such wall-motion induced CSF flow has been modeled and
simulated successfully in the human brain ventricular system~\cite{Causemann2022HumanFramework,
Kurtcuoglu2005ComputationalSystem, Kurtcuoglu2007ComputationalSylvius, Linninger2005PulsatileBrain}.

\subsection*{Estimation of computational model parameters}
The diffusion coefficients used in this study were partially based on values
from experiments and partially estimated. The value $D_3=\num{1.15e-10} \ \mathrm{m^2/s}$
is based on the value $\num{1.15e-10} \pm 0.11 \ \mathrm{m^2/s}$ reported in an
experiment~\cite{GuraSadovsky2017MeasurementExpansion}, while the values $D_2$ and
$D_1$ were extrapolated based on experimental data~\cite{Swaminathan1997PhotobleachingDiffusion,
Potma2001ReducedCells, Mahmood2023ExosomeTemperature} using the Stokes-Einstein
relation~\cite{Einstein1905UberTeilchen}~(\Cref{eq:D_stokes_einstein}).
This extrapolation assumes the diffusing substance is of spherical shape, which proteins are not,
and the relation does not necessarily hold in CSF. Consequently, there is uncertainty in the values used.
We note that Sadovsky~\emph{et al.}~\cite{GuraSadovsky2017MeasurementExpansion} reported that
Dendra2 satisfies the Stokes-Einstein in the solution they used, a solution which had similar viscosity
to water. We used CSF density and viscosity values measured in humans~\cite{Bloomfield1998EffectsFluid},
and because embryonic CSF in zebrafish is high in proteins, nutrients and other substances,
the CSF in which photoconverted Dendra2 are transport in our experiments may be more viscous
than what we consider. This would reduce the diffusion coefficient~(cf.~\Cref{eq:D_stokes_einstein}),
but based on reports of protein concentration having a limited impact on the viscosity of
CSF~\cite{Bloomfield1998EffectsFluid, Brydon1995PhysicalViscosity}, this reduction is presumably
limited. Merely the presence of other proteins and nutrients may also reduce the diffusion
coefficient by reduction of mean-free path lengths of molecules~\cite{Cussler2009Diffusion:Systems}.
Considering the diffusion coefficients $D_1$ and $D_2$ both smaller than $D_3$ in our transport analysis
provides information about how the simulation results of Dendra2 transport would be altered if the
diffusion coefficient $D_3$ is overestimated. In the case that $D_3$ is underestimated, we assume the
results would follow the same trend as the differences between results for $D_2$ and $D_3$: diffusion becomes
more important, and is almost balancing advective transport for $D_3$. Thus for a higher diffusion coefficient,
we would see transport that is mostly explained by diffusive behavior.

We heuristically determined the parameter $\tau = \num{6.5e-4}$ Pa, which determines
the magnitude of the cilia tangential traction boundary conditions. The maximum force applied
was in the dorsal region of the middle ventricle, where the posterior region had a tangential
traction vector magnitude of $\vert\btau\vert_{\mathrm{max}} = \num{2.5e-3}$ Pa applied to the
boundary. A couple of previous simulation studies of cilia-mediated flow in brain ventricles have
used a volumetric force density to represent the cilia. Siyahhan~\emph{et al.}~\cite{Siyahhan2014FlowVentricles}
used a force expression that increased linearly from the wall to a maximum force density of
$f_{\mathrm{max}} = 526$ N/$\mathrm{m^3}$ at a distance 15 $\mu$m away from the wall,
which represented cilium length. If we assumed a cilia array of the same thickness
$l = 15$ $\mu$m in our model, our maximum traction parameter $\vert\btau\vert_{\mathrm{max}}$
would correspond to a maximum force density of 
\begin{equation*}
    f_c = \frac{\vert\btau\vert_{\mathrm{max}}}{l} = \frac{\num{2.5e-3} \ \mathrm{Pa}}{15 \ \mu\mathrm{m}} = 169 \ \mathrm{N/m^3},
\end{equation*}
which smaller than $f_{\mathrm{max}}$, albeit the same order of magnitude.
The length 15 $\mu m$ is the a typical length of cilia in human brain ventricles
(in general the length is between 10--15 $\mu$m~\cite{Afzelius2004Cilia-relatedDiseases}),
while in zebrafish they are closer to 5 $\mu$m~\cite{Salman2022ComputationalEmbryo}.
With $l=5$ $\mu$m, our traction parameter corresponds to a maximum force density of
$f_c=507$ N/$\mathrm{m^3}$, which is in good agreement with $f_{\mathrm{max}} = 526$ N/$\mathrm{m^3}$
used by Siyahhan~\emph{et al.}~\cite{Siyahhan2014FlowVentricles}.
We note that they based $f_{\mathrm{max}}$ on reconstructions of flow velocities in murine brain ventricles.

Another study that used a force density to model ciliary motion is
Thouvenin~\emph{et al.}~\cite{Thouvenin2020OriginCanal}, where two-dimensional flow
in the central canal of zebrafish embryos was modeled using a cilia force density of
4000 N/$\mathrm{m^2}$. The resulting numerical velocity fields matched well with
\emph{in vivo} measurements, where the maximum velocity magnitude was around 5 $\mu$m/s
(recall that our model exhibits a maximum of 27.7 $\mu$m/s).
The geometry they studied resembles a pipe, probably offering more flow resistance
than the open geometry considered in our work, which may be why they needed a larger
force density to accurately represent experimental results. In any case, it is
interesting to note that our and these two previous studies have estimated
cilia force densities that only differ by one order of magnitude.

\subsection*{Numerical approximation and convergence rates}
The Brezzi-Douglas-Marini and discontinuous Galerkin discretization scheme
($\mathrm{BDM}_1$-$\mathrm{DG}_0$) that we employ for the Stokes equations
is computationally more expensive than a more conventional approach based on
e.g. Taylor-Hood discretization~\cite{Stenberg1990ErrorProblem}
($\mathbb{P}_2-\mathbb{P}_1$ elements) or lower-order enriched elements~\cite{Brezzi2011MixedMethods},
such as the mini-element ($\mathbb{P}_1-\mathbb{P}_1$ with a velocity stabilization).
Here, $\mathbb{P}_k$ is the set of continuous Lagrange polynomials of order $k$.
We opted for the BDM-DG scheme because of its exactly-mass-conserving property,
since numerical experiments proved that a divergence-free velocity field was
paramount to attain a stable discretization of the advection-diffusion equation
in advection-dominated regions. Preliminary attempts using Taylor-Hood elements
proved hard to achieve stable transport results with a reasonable computational mesh size.
Stability either required excessive mesh refinement and/or problem-specific tuning of
hyperparameters when employing a Streamline-Upwind/Petrov Galerkin stabilization
method~\cite{Franca1992StabilizedModel}. Probing numerical values of the divergence of
the velocity when using Taylor-Hood elements revealed that, on the standard computational
mesh, the $L^2$ norm of $\nabla\cdot\uu_h$ was in the order of $10^{-6}$, while the
maximum value of the divergence of the velocity was of order unity. This is a result
of the fact that incompressibility is only achieved in a distributional sense when
employing Taylor-Hood elements. On a refined mesh with roughly six times as many
computational cells as the standard mesh, the numerical values of
$\normltwo{\nabla\cdot\uu_h}$ and $\normlinf{\nabla\cdot\uu_h}$ were still
of the same orders as reported for the standard mesh. This may lead to the
hyperbolic advective term in the advection-diffusion equation accumulating errors
that grow unstable over time. 

In addition to its favorable mass-conserving property, the BDM-DG scheme
lets us impose the impermeability condition strongly, and the tangential
traction boundary condition weakly, without any tailored approach,
since these boundary conditions appear naturally in the weak form of the
Stokes equations~(\cref{subsec:appendixA1}). If instead e.g. Taylor-Hood
elements were to be used, the impermeability boundary condition must
be handled with for example a Nitsche method~\cite{Nitsche1971UberSind}
or with a Lagrange multiplier~\cite{Babuska1973TheMultipliers, Bertoluzza2017BoundaryHemodynamics}. 

\emph{A priori} error estimates for the $\mathrm{BDM}_k$-$\mathrm{DG}_{k-1}$ scheme are naturally 
derived in a mesh-dependent broken $H^1$-norm for $\uu$ and the $L^2$-norm for the pressure. The respective 
errors are expected to decay with order $k$, see for example Cockburn~\emph{et al.}~\cite{Cockburn2005AEquations}. For the sake of simplicity,~\Cref{tab:cylinder_error_rates} reports the velocity errors in the standard $H^1$ norm. Using polynomial degree  
$k=1$, we observe linear convergence in both the velocity and the pressure.   
\mk{MK: I see 1.5 in the table for $\uu$ in $L^2$?} \lyng{HH: Yes, while the $H^1$ error decays linearly, isn't this in line with expectations? Could be more specific: "Using polynomial degree 
$k=1$, we observe linear convergence in both the velocity $H^1$ error and the pressure $L^2$ error."}

\lyng{HH: Add a paragraph discussing the advection-diffusion equation discretization with DG elements?}

\subsection*{The role of motile cilia in intraventricular flow and transport}
Our work has studied CSF flow and solute distribution
in a relatively small and simple model, which is devoid of
active CSF secretion and CSF absorption. Even though there is
a clear conservation of the ventricular system across vertebrates,
there are also important considerations to take, which include
ventricular size and geometries, properties of ciliated cells,
animal posture, and magnitude of velocities. For instance, a large
volume would require an extensive amount of time for solutes to distribute
if based solely on diffusion. Besides, the presence of multiple cilia
instead of a single cilium per cell, as described in mammalian ependymal cells,
will generate much stronger velocities. Hence advection may be more
dominant in those situations. However, in large brains, cilia beating will
only contribute to a restricted flow at the very close proximity of the ventricular walls,
while blood vessel pulsations, respiration and CSF secretion will generate a
pulsatile bulk flow in the central regions of the ventricles~\cite{Siyahhan2014FlowVentricles}.
Based on this, cilia probably play an important role in regulating the
amount of substances close to the ventricular walls, similar to coral
reefs, where it has been observed that cilia are primary regulators of
oxygen amounts at the coral surface~\cite{Pacherres2022CiliaryProduction}.
But for zebrafish embryos, the ventricles are so small that the flow
generated by ciliary motion is of the scale of the geometry, and thus not
only contributes to near-wall flow dynamics, but the whole ventricular system.
As of now, motile cilia deficiency has been associated with altered brain function
in the zebrafish and ventricular enlargement in zebrafish, rodents and humans.
Yet the molecular mechanisms leading to these alterations are poorly understood.
We trust that our model sets the basis for future work addressing these open questions.
We are also confident that our model can be adjusted and implemented to fit
microscopic and macroscopic parameters once acquired for larger brains.

\lyng{HH: Refer to Eichele et al. article discussion about role of cilia.}


\subsection*{Limitations and further work}
The model presented has certain limitations. Future work could consider:
\begin{itemize}
    \item spatial heterogeneity of the cilia, how this affects flow and transport
    \item time-varying cilia?
    \item improved advection-diffusion equation BCs
    \item Modeling deformation of the ventricles and investigating how the arising flow compares with pulsatile flow observed in experiments
\end{itemize}

\section*{Conclusion}
We have presented experimental data from imaging photoconversion of a fluorescent protein in zebrafish brain ventricles, both for healthy animals and mutant animals with ciliary dysfunction. Experimental results indicate that cilia only affect the local distribution of proteins, while most of the transport is by diffusion. We also presented a numerical model of flow and transport in zebrafish brain ventricles based on finite element methods. A realistic brain ventricles geometry from medical imaging was used to perform a computational study with the presented model. The flow model reproduces qualitatively the experimentally observed flow features of cilia-mediated cerebrospinal fluid flow in zebrafish brain ventricles. Coupling the flow to the transport model, we estimated how slowly or rapidly molecules of varying size are transported in the ventricles. Simulations indicate that cilia-driven advection is important for transport of larger molecules.


%%--------------ACKNOWLEDGMENTS--------------%%
\section*{Acknowledgments}
Halvor Herlyng acknowledges support from the national infrastructure for computational
science in Norway, Sigma2, via grant \#NN8049K. Thanks are extended to J\o rgen Dokken
for valuable model implementation discussions.
This work was funded by the Research Council of Norway (NJY FRIPRO grants 314189 and 353348).
\newpage


%%--------------APPENDIX--------------%%
\appendix
\section{Supplementary Theory}
\subsection{Weak formulation of the Stokes equations}\label{subsec:appendixA1}
We consider the time-dependent Stokes equations
\begin{subequations}
\begin{alignat}{2}
   \rho\pdifft{\uu} - \nabla \cdot \bsig(\uu, p) &= \mathbf{0} \quad &&\mathrm{in} \ \Omega, \label{eq:BDM_stokes_mom}\\
  \nabla \cdot \uu &= 0 \quad &&\mathrm{in} \ \Omega, \label{eq:BDM_stokes_div} 
\end{alignat}
\label{eq:appendix_stokes_eqs}% 
\end{subequations}%
in the domain $\Omega$ with boundary $\Gamma$, where
$\bsig(\uu, p) = 2\mu\beps(\uu) - p\mathbf{I}$ is the viscous
stress tensor, in which $\beps(\uu) = \frac{1}{2}(\nabla\uu + (\nabla\uu)^T)$
is the symmetric gradient. Furthermore, $\uu\in\VV$ is the fluid velocity
and $p\in Q$ is the fluid pressure, and $\rho$ and $\mu$ are the fluid density
and dynamic viscosity, respectively. Unless otherwise stated, we let $\bsig = \bsig(\uu, p)$. 

We will consider the function spaces $\VV$ and $Q$ based on the
Sobolev spaces $\mathbf{H}^1(\mathrm{div}; \Omega)$ and $L^2(\Omega)$,
respectively, defined as 
\begin{align*}
    L^2(\Omega) &= \bigg\{p \ : \ \intO{p^2} < \infty\bigg\} \\
    \mathbf{H}^1(\mathrm{div}; \Omega) &= \bigg\{\uu \ : \ \uu\in\mathbf{L}^2(\Omega), \nabla\cdot\uu\in L^2(\Omega)\bigg\}.
\end{align*}
Bold-faced characters for a space denotes a $d$--dimensional vector space,
where $d$ is the spatial dimension of $\Omega$. For example,
$\mathbf{H}^1(\Omega) = [H^1(\Omega)]^3$ if $\Omega$ is three-dimensional.
We will consider $p\in Q=L^2_0(\Omega)$, with subscript 0 meaning
that $\intO{p} = 0$, and $\uu\in\VV$ with
\begin{equation*}
    \VV =  \bigg\{\vv \ : \ \vv\in\mathbf{H}^1(\mathrm{div}; \Omega), \vv\cdot\nn=0 \ \mathrm{on} \ \Gamma\setminus\Gamma_{\mathrm{io}}\bigg\}.
\end{equation*}
Here $\Gamma_{\mathrm{io}}$ denotes an in- or outflow boundary. 

\mk{
You should be careful here. The H(div) functions do not have a
well defined gradient (and in turn $\varepsilon$). However, for the derivations of
the continuous problem you don't need H(div). I would derive all the weak forms
in $V = \left\{ v\in H^1, v\cdot n = \text{on} \Gamma\right\}$. Then in A.2 you say we
chose spaces $V_h$ which are are not conforming, i.e. it does not hold that $V_h\subset V$.
Instead you have $V_h\subset H(div)$. Reasons is better mass csrv and as a
bonus the bcs can be enforced strongly. Note that H(div) functions can have
jumps in tangential components on surfaces in $\Omega$. Then introduce the discrete weak form.  
}
To derive a weak formulation of the Stokes equations~\eqref{eq:appendix_stokes_eqs},
we begin by multiplying~\eqref{eq:BDM_stokes_mom} with a
test function $\vv\in\VV$ and integrating over the domain:
\begin{equation*}
    \intO{\rho\pdifft{\uu}\cdot\vv} - \intO{(\nabla\cdot\bsig)\cdot\vv} = \mathbf{0}.
\end{equation*}
Performing integration by parts of the stress term and applying Green's first identity, we get
\begin{equation}
    \intO{\rho\pdifft{\uu}\cdot\vv} - \intG{\bsig\nn\cdot\vv} + \intO{2\mu\beps(\uu) : \beps(\vv)} - \intO{p\nabla\cdot\vv} = \mathbf{0}, \label{eq:modelA_weak_form1}
\end{equation}
where the symbol : signifies the matrix inner product,
$A:B=\text{trace}{(A^TB)}$ for matrices $A, B\in\mathbb{R}^{d\times d}$. 

Further steps depend on the boundary conditions applied,
which is different for the three model versions we consider.
Before going through the different model versions, let us introduce
some notation used in defining slip boundary conditions. We make use of a decomposition
of any vector $\vv$ on the boundary $\Gamma$ into its normal and tangential components,
$\vv = \vv_{\perp} + \vv_{\parallel}$, where $\vv_{\perp}=(\vv\cdot\nn)\nn$ and
$\vv_{\parallel}=P_{\nn}(\vv)$. The tangential projection is defined as
\begin{equation}
    P_{\nn}(\vv) = (\mathbf{I} - \nn\otimes\nn)\vv,
    \label{eq:projection_operator_appendix}
\end{equation}
where $\nn$ is the outer unit normal of the surface. Introducing
$\hat{\bsig} = \bsig\nn$ for the traction (vector), the forcing due
to cilia is represented by constraining $\bsigpar$ on the part of the
boundary populated by cilia. In particular, we let
$\bsigpar = \btau(\xx, \rr) = \tau\lambda(\xx) P_{\nn}(\rr)$,
where the parameter $\tau=\num{6.5e-4}$ Pa was used to control the magnitude
of the tangential stresses, and chosen by calibrating simulated velocity
fields with experimental data~\cite{Olstad2019CiliaryDevelopment}.
The details of the function $\lambda(\xx)$ follow subsequent to deriving the weak form.

We choose $\mathbf{r} = \rr^+ = (1, 0, 1)$ or $\mathbf{r} = \rr^- = -(1, 0, 1)$ to impose forces perpendicular to the $y$-axis.
The reason for this is that all of the experimental images we compare with are taken in the $xz$-plane,
rendering flow features in the $y$-direction unobservable.
Next, we consider the way of handling the stress boundary integral for each model version. 

\textbf{Only-cilia/no-cardiac model.} This model version has cilia forces
as the only mechanism driving flow. We consider the
Stokes equations~\eqref{eq:appendix_stokes_eqs} with the slip boundary conditions
\begin{subequations}
    \begin{alignat}{2}
        \uu\cdot\nn &= 0 \quad &&\mathrm{on} \ \Gamma=\Gc\cup\Gs, \label{eq:modelA_slip_BC_impermeability} \\
        %\hat{\sigma} - (\hat{\sigma}\cdot\nn)\nn &= \tau P_{\nn}(\rr) \quad &&\mathrm{on} \ \Gamma_s,
        \bsigpar &= \btau \quad &&\mathrm{on} \ \Gc,
        \label{eq:modelA_slip_BC_shear} \\
        \bsigpar &= \mathbf{0} \quad &&\mathrm{on} \ \Gs,\label{eq:modelA_full_slip_BC}
    \end{alignat}\label{eq:modelA_slip_BC}%
\end{subequations}%
on the boundary $\Gamma=\Gamma_c\cup\Gamma_s$. The boundary $\Gamma_s$ is a free-slip boundary, where we impose the free-slip condition~\eqref{eq:modelA_full_slip_BC}. The first condition~\eqref{eq:modelA_slip_BC_impermeability} is the impermeability condition, enforcing no flow normal to the boundary, whereas~\eqref{eq:modelA_slip_BC_shear} imposes a stress tangential to the boundary. These tangential stresses are applied to represent the forces that the collective cilia motion exerts on the cerebrospinal fluid. 

The boundary conditions~\eqref{eq:modelA_slip_BC} are set weakly through the stress boundary integral 
\begin{equation}
    \intG{\bsig\nn\cdot\vv} =  \intG{\hat{\bsig}\cdot\vv}
    \label{eq:boundary_integral_BC1}
\end{equation}
in the weak form~\eqref{eq:modelA_weak_form1} of the Stokes equations. We decompose the traction vector $\hat{\bsig}$ into a normal and tangential component:
\begin{equation*}
    \hat{\bsig} = \bsigperp + \bsigpar
\end{equation*}
Inserting the above into~\eqref{eq:boundary_integral_BC1}, combined with application of the tangential traction condition~\eqref{eq:modelA_slip_BC_shear} and the free-slip condition~\eqref{eq:modelA_full_slip_BC}, yields
\begin{equation}
    \intG{\hat{\bsig}\cdot\vv} = \int_{\Gc\cup\Gs}{(\bsigperp + \bsigpar)\cdot\vv}\, ds = 
    \int_{\Gc\cup\Gs}{\bsigperp\cdot\vv}\, ds + \intGc{\btau\cdot\vv} = \intG{\bsigperp\cdot\vv} + \intGc{\btau\cdot\vv}.
    \label{eq:boundary_integral_BC2}
\end{equation}
The second term on the right-hand side of~\eqref{eq:boundary_integral_BC2} is moved to the right-hand side of the weak form, since $\btau$ is known. Using $\hat{\bsig}_{\perp} = (\hat{\bsig}\cdot\nn)\nn$, we can write the first term on the right-hand side of~\eqref{eq:boundary_integral_BC2} as
\begin{equation}
    \intG{\bsigperp\cdot\vv} = \intG{(\hat{\bsig}\cdot\nn)\nn\cdot\vv} = \intG{(\hat{\bsig}\cdot\nn)(\vv\cdot\nn)}.
    \label{eq:boundary_integral_BC_normal}
\end{equation}
In a later section, we will discretize the weak formulation of the Stokes equations with Brezzi-Douglas-Marini finite elements~\cite{Brezzi1985TwoProblems}. These elements are defined through integral moments of the normal component of a variable on the facets of an element, we can strongly enforce the value of $\uu\cdot\nn$ and thus the boundary condition~\eqref{eq:modelA_slip_BC_impermeability}. Consequently, the test function counterpart $\vv\cdot\nn$ vanishes on the boundary, and the term~\eqref{eq:boundary_integral_BC_normal} is therefore dropped from the weak form. As a result, for model verison I, the weak form of the momentum equation takes the following form:
\begin{equation*}
    \intO{\rho\pdifft{\uu}\cdot\vv} + \intO{2\mu\varepsilon(\uu) : \varepsilon(\vv)} - \intO{p\nabla\cdot\vv} = \intGc{\btau\cdot\vv}, \quad\forall\vv\in\VV.
\end{equation*}
Next, multiplying the continuity equation~\eqref{eq:BDM_stokes_div} with a test function $q\in Q$, we require that
\begin{equation}
    -\intO{q\nabla\cdot\uu} = 0,\quad\forall q\in Q
    \label{eq:pressure_weak_form}
\end{equation}
The full weak form of the Stokes equations for model version I is then: find $(\uu, \, p) \in \VV\times Q$, such that
\begin{subequations}
    \begin{alignat}{5}
        &\intO{\rho\pdifft{\uu}\cdot\vv} + &\intO{2\mu\varepsilon(\uu) : \varepsilon(\vv)} - &\intO{p\nabla\cdot\vv} &&= \intGc{\btau\cdot\vv}, &&\quad\forall\vv\in\VV, \label{eq:stokes_weak_mom_modelA}\\ 
        &{} &{}  -&\intO{q\nabla\cdot\uu} &&= 0, &&\quad\forall q\in Q. \label{eq:stokes_weak_cont_modelA}
    \end{alignat}%
    \label{eq:stokes_weak_modelA}%
\end{subequations}%
We observe that, in this case with slip boundary conditions on the entirety of the boundary $\Gamma$, the
vector $(\uu, p)=(\mathbf{0}, C)$ for any real constant $C$ satisfies the equations.
The system~\eqref{eq:stokes_weak_modelA} is thus singular with a one-dimensional nullspace of constant pressures.
Consequently, assembly of our discretization of the  equations~\eqref{eq:stokes_weak_modelA} into a
linear equation system $\mathbf{A}\xx=\mathbf{b}$ yields a singular matrix $\mathbf{A}$.
To ensure existence of solutions, we orthogonalize the right-hand side vector $\mathbf{b}$ with
respect to the pressure nullspace basis (the set of all constants), before solving the linear system.
To obtain a unique solution, which together with existence renders the problem well-posed,
information of the pressure nullspace is passed to the linear solver. Let us remark that
the singularity is a consequence of the fact that the impermeability condition is assumed
to hold on the \emph{entire} domain boundary for this model version, which leaves the
pressure under-determined as a result of no boundary condition specifying the pressure.

The tangential stresses $\btau = \tau\lambda(\xx) P_{\nn}(\rr)$ that represent the cilia
motion differ for the cilia regions. We heuristically determined $\tau$ and $\lambda(\xx)$.
As noted previously, $\tau=\num{6.5e-4}$ Pa. We set $\rr = \rr^+$ and $\rr = \rr^-$ in
the ventral and dorsal regions of the diencephalic ventricle, respectively. In the ventral region,
we set the linear profile $\lambda(\xx) = \frac{1}{2}\left(1 - \frac{x-x_{0, v}}{x_{e, v} - x_{0, v}}\right)$,
where $[x_{0, v}, x_{e, v}]$ is the $x$-range of the cilia region, so the stresses decrease
linearly towards the posterior part of the geometry. We impose a linearly increasing profile
over the $x$-range $[x_{0, d}, x_{e, d}]$ in the dorsal region:
$\lambda(\xx)=\frac{11}{4}\left(\frac{x-x_{0, d}}{x_{e, d} - x_{0, d}}\right)$.
\lyng{Put all of this information on $\lambda$ choice in a table.}

In the telencephalic ventricle, we set $\rr=\rr^-$ and $\lambda=\frac{2}{5}$
in the anterior most part of the cilia region,
and $\rr=\rr^+$ and $\lambda(\xx)=1$ in the posterior part.

\textbf{Only-cardiac/no-cilia model.} For this model version,
the only physical mechanism inducing fluid flow is the cardiac cycle.
We model the influence of the cardiac cycle with a normal pressure boundary condition:
\begin{subequations}
    \begin{alignat}{2}
      \uu\cdot\nn = 0, \ \hat{\bsig}_{\parallel} &= \mathbf{0} \quad &&\mathrm{on} \ \Gs, \label{eq:modelB_slip_BC} \\
      (\mu\nabla\uu - p\mathbf{I})\nn &= \Tilde{p}(t)\nn \quad &&\mathrm{on} \ \Gp, \label{eq:modelB_pressure_BC}
    \end{alignat}%
\end{subequations}%
where the boundary $\Gamma = \Gs\cup\Gp$ consists of a free-slip boundary $\Gs$, and a part $\Gp$ where a normal traction type of boundary condition is imposed. Note, however, that the transpose velocity gradient term must be included inside the parentheses on the left-hand side of~\eqref{eq:modelB_pressure_BC} for it to be the traction $\bsig$. Since this is not the case, we rather refer to this boundary condition as a normal pressure boundary condition. The boundary $\Gs$ is handled as discussed in the previous section, such that the stress boundary integral~\eqref{eq:boundary_integral_BC1} vanishes over $\Gs$. Therefore, we now have
\begin{align}
    \intG{\hat{\bsig}\cdot\vv} = \intGp{\hat{\bsig}\cdot\vv} &= \intGp{(\mu(\nabla\uu + (\nabla\uu)^T) - p\mathbf{I})\nn\cdot\vv}\notag \\
    &= \intGp{\mu(\nabla\uu)^T\nn\cdot\vv} + \intGp{(\mu\nabla\uu - p\mathbf{I})\nn\cdot\vv}\notag\\
    &= \intGp{\mu(\nabla\uu)^T\nn\cdot\vv} + \intGp{\Tilde{p}(t)(\vv\cdot\nn)}
    \label{eq:modelB_boundary_integral1}
\end{align}
We leave the first term on the left-hand side of~\eqref{eq:modelB_boundary_integral1} in the weak form of the momentum equation. The second term is determined by the boundary condition~\eqref{eq:modelB_pressure_BC}, and is therefore moved to the right-hand side of the weak form. The weak form of the momentum equation for model version IIa is then
\begin{equation}
    \begin{aligned}
    \intO{\rho\pdifft{\uu}\cdot\vv} + \intO{2\mu\varepsilon(\uu) : \varepsilon(\vv)} - \intO{p\nabla\cdot\vv} &-\intGp{\mu(\nabla\uu)^T\nn\cdot\vv} \\ &=\intGp{\Tilde{p}(t)(\vv\cdot\nn)}, \quad\forall\vv\in\VV.
    \end{aligned}
    \label{eq:weak_form_mom_eq_modelB}
\end{equation}
Also considering the weak form of the continuity equation, the weak formulation for model version II then reads: find $(\uu,\, p)\in\VV\times Q$ such that~\eqref{eq:weak_form_mom_eq_modelB} and~\eqref{eq:pressure_weak_form} hold for all $(\vv, \, q)\in\VV\times Q$.
To simulate the back-and-forth motion induced by the cardiac cycle observed in previous work~\cite{Olstad2019CiliaryDevelopment}, we apply sinusoidal forcing. The boundary $\Gamma_p$ has two separate regions: one located on the anterior ventricle and one on the posterior ventricle~(\cref{fig:figure1}, panel \textbf{b}). For the normal pressure boundary condition, we set $\Tilde{p}(t) = 0$ on the anterior part and
$\Tilde{p}(t) = p_c(t) = -A\sin{\omega t}$ on the posterior part. The value of the amplitude $A$ was based on heuristics and $\omega = 2\pi f$, where $f=2.22$ Hz is the cardiac cycle frequency. 

\textbf{Baseline model.} This model version, regarded as the baseline model, considers both cilia and cardiac beating as driving mechanisms of fluid flow. We therefore consider the boundary conditions
\begin{subequations}
    \begin{alignat}{2}
      \uu\cdot\nn = 0, \ \bsigpar &= \btau  \quad &&\mathrm{on} \ \Gc, \label{eq:slip_BC_modelC} \\
      \uu\cdot\nn = 0, \ \bsigpar &= \mathbf{0}  \quad &&\mathrm{on} \ \Gs, \label{eq:freeslip_BC_modelC} \\
      (\mu\nabla\uu - p\mathbf{I})\nn &= \Tilde{p}(t)\nn \quad &&\mathrm{on} \ \Gp, \label{eq:pressure_BC_modelC}
    \end{alignat}
\end{subequations}%
where $\Gamma = \Gc\cup\Gs\cup\Gp$ is the boundary of $\Omega$. The quantities $\btau$ and $\Tilde{p}(t)$ are as introduced in the sections on model versions I and IIa. We handle the boundary conditions~\eqref{eq:slip_BC_modelC} and~\eqref{eq:freeslip_BC_modelC} as introduced for model version I, and the boundary condition~\eqref{eq:pressure_BC_modelC} is applied as discussed in the section on model IIa. The abstract weak formulation for model version 0 is then: find $(\uu,\, p)\in\VV\times Q$ such that
\begin{subequations}
    \begin{alignat}{2}
        \mathcal{A}(\uu, \vv) + \mathcal{B}(\vv, p) &= \mathcal{L}(\vv), &&\quad\forall\vv\in\VV, \\
        \mathcal{B}(\uu, q) &= 0, &&\quad\forall q\in Q,
    \end{alignat}%
    \label{eq:abstract_weak_form_modelC}
\end{subequations}%
with the definitions
\begin{align}
    \mathcal{A}(\uu, \vv) &= \intO{\rho\pdifft{\uu}\cdot\vv} + \intO{2\mu\varepsilon(\uu) : \varepsilon(\vv)} -\intGp{\mu(\nabla\uu)^T\nn\cdot\vv}, \label{eq:stokes_bilinear_form}\\
    \mathcal{B}(\uu, q) &= -\intO{q\nabla\cdot\uu}, \label{eq:stokes_skew_form}\\
    \mathcal{L}(\vv) &= \intGc{\btau\cdot\vv} + \intGp{\Tilde{p}(t)(\vv\cdot\nn)}. \label{eq:stokes_linear_form}
\end{align}

\subsection{Discretization of the Stokes equations}
We will outline the discretization of the weak formulation~\cref{eq:abstract_weak_form_modelC,eq:stokes_bilinear_form,eq:stokes_skew_form,eq:stokes_linear_form} of the Stokes equations for model version 0. A similar approach can be used to discretize the weak formulation for the other model versions. First, we consider the spatial discretization of the weak formulation. We rewrite the weak form in terms of the discretized variables, denoted with the subscript $h$:
\begin{subequations}
    \begin{alignat}{2}
        \mathcal{A}(\uu_h, \vv_h) + \mathcal{B}(\vv_h, p_h) &= \mathcal{L}(\vv_h), &&\quad\forall\vv_h\in\VV_h, \\
        \mathcal{B}(\uu_h, q_h) &= 0, &&\quad\forall q_h\in Q_h.
    \end{alignat}%
    \label{eq:no_stabilization_abstract_discrete_weak_form_modelC}%
\end{subequations}%
Our choice for the discrete spaces $\VV_h$ and $Q_h$ are finite elements of the Brezzi-Douglas-Marini family~\cite{Brezzi1985TwoProblems} and discontinuous Galerkin elements, respectively. Application of this scheme requires a stabilization of the bilinear form $\mathcal{A}(\uu_h, \vv_h)$. We apply the stabilization outlined in Hong~\emph{et al.}~\cite{Hong2016AEquations}. First, we define the average and jump operators, respectively, as
\begin{equation*}
    \avg{\uu} = \frac{\uu^+\cdot\nn^+ - \uu^-\cdot\nn^-}{2} \qquad\mathrm{and}\qquad \jump{\uu} = \uu^+ - \uu^-,
\end{equation*}
where $\nn$ is the normal vector on a facet separating two cells, where the plus and minus signs denote variables in each of the two cells. The normal vector is chosen such that it points towards the exterior of its corresponding cell. On the boundary, the operators are defined as $\avg{\uu} = \uu\cdot\nn$ and $\jump{\uu} = \uu$,
with $\nn$ pointing to the exterior of the domain. We now define and add the bilinear form
\begin{align*}
    \mathcal{A}_{\mathrm{stab}}(\uu_h, \vv_h) &= \sum_{k\in K}\intF{2\mu\frac{\gamma}{\avg{h}}\jump{P_{\nn}(\uu_h)}\cdot \jump{P_{\nn}(\vv_h)}}\\
    &-\sum_{k\in K}\intF{2\mu\avg{\beps(\uu_h)}\cdot\jump{\vv_h}}-\sum_{k\in K}\intF{2\mu\avg{\beps(\vv_h)}\cdot\jump{\uu_h}}
\end{align*}
to the discrete weak  formulation~\eqref{eq:no_stabilization_abstract_discrete_weak_form_modelC}, where $\mathcal{F} = \{\mathcal{F}_k : k\in K\}$ is the set of all interior facets $\mathcal{F}_k$ of the computational mesh. The parameter $\gamma>0$ is introduced to penalize jumps in the tangential components of the velocity across facets, wheras $h$ is the cell diameter of a mesh cell, defined as the maximal distance between two points in a cell. The semi-discrete weak formulation of the Stokes equations reads: find ($\uu_h, p_h$)$\in$($\VV_h, Q_h$) such that
\begin{subequations}
    \begin{alignat}{2}
        \mathcal{A}(\uu_h, \vv_h) + \mathcal{A}_{\mathrm{stab}}(\uu_h, \vv_h) + \mathcal{B}(\vv_h, p_h) &= \mathcal{L}(\vv_h), &&\quad\forall\vv_h\in\VV_h, \\
        \mathcal{B}(\uu_h, q_h) &= 0, &&\quad\forall q_h\in Q_h.
    \end{alignat}%
    \label{eq:abstract_semi_discrete_weak_form_modelC}%
\end{subequations}%
We solved~\eqref{eq:abstract_semi_discrete_weak_form_modelC} numerically with the penalty parameter $\gamma=10$.

In time, we discretize the weak formulation~\eqref{eq:abstract_semi_discrete_weak_form_modelC} with the implicit Euler method.

\subsection{Weak formulation of the advection-diffusion equation}\label{subsec:adv_diff_eq_weak_form}
Recall the strong form of the advection-diffusion equation:
\begin{equation}
    \pdifft{c} + \nabla\cdot\JJ = 0 \quad\mathrm{in} \ \Omega,
    \label{eq:adv_diff_strong_DG}
\end{equation}
where $\Omega$ is the domain with boundary $\Gamma$, and the total solute flux
\begin{equation*}
    \JJ(c) = c\uu - D\nabla c.
\end{equation*}
We consider $\JJ = \JJ(c)$, unless otherwise indicated.
The concentration $c\in W$ and the cerebrospinal fluid velocity $\uu\in\VV$.
The weak form of~\eqref{eq:adv_diff_strong_DG} is derived by
multiplying the equation with a test function $\phi \in W$ and integrating over the domain: 
\begin{equation*}
    \intO{\pdifft{c}\phi} + \intO{(\nabla\cdot\JJ)\phi} = 0.
\end{equation*}
Integrating the second term by parts and applying Green's first identity yields
\begin{equation}
    \intO{\pdifft{c}\phi} + \intG{(\JJ\cdot\nn)\phi} - \intO{\JJ\cdot\nabla\phi} = 0.\label{eq:adv_diff_weak_formA1}
\end{equation}
The boundary integral of the total flux is handled differently for the three different model versions, depending on the boundary conditions considered.

\textbf{Only-cilia/no-cardiac model.} We consider $\Omega$ to be closed and with impermeable walls, setting $\JJ\cdot\nn=0$ on $\Gamma$. The weak form is then reduced to
\begin{equation*}
    \intO{\pdifft{c}\phi} - \intO{\JJ\cdot\nabla\phi} = 0, \quad\forall\phi\in W.
\end{equation*}
Defining the left-hand side as the bilinear form
\begin{equation*}
    a(c, \phi) = \intO{\pdifft{c}\phi} - \intO{\JJ\cdot\nabla\phi}, 
\end{equation*}
we have the following abstract weak formulation for model version I: find $c\in W$ such that
\begin{equation*}
    a(c, \phi) = 0, \quad\forall\phi\in W.
\end{equation*}

\textbf{Only-cardiac/no-cilia model.} When using the normal pressure boundary condition~\eqref{eq:modelB_pressure_BC} to model pulsatile flow,
we have in- and outflow of cerebrospinal fluid and thus advection of $c$ on the pressure boundary $\Gp$.
Let $\Gp=\Gin\cup\Gout$ and $\Gamma = \Gnf\cup\Gin\cup\Gout$, with 
\begin{align*}
    \Gin  &= \bigg\{\xx\in\Gamma : \uu(\xx, t)\cdot\nn(\xx) < 0 \bigg\}, \\
    \Gout &= \bigg\{\xx\in\Gamma : \uu(\xx, t)\cdot\nn(\xx) > 0 \bigg\},
\end{align*}
as in- and outflow boundaries, respectively, where we have emphasized that
both $\uu$ and $\nn$ are spatially dependent variables.
With $\JJ\cdot\nn=0$ on the no-flux boundary $\Gnf$, we have
\begin{equation*}
    \intG{(\JJ\cdot\nn)\phi} = \intGin{(\JJ\cdot\nn)\phi} + \intGout{(\JJ\cdot\nn)\phi}. 
\end{equation*}
For the outflow boundary integral over $\Gout$, we impose zero diffusive flux $D\nabla c\cdot\nn=0$,
with the advection term $c(\uu\cdot\nn)$ simply left in the variational form as an unknown.
For the continuous problem this is problematic, since we are lacking a boundary condition to close the problem. 
In the discrete setting, however, this method of leaving the boundary term has been shown
to yield consistent results~\cite{Papanastasiou1992ACondition, Lynch2020NumericalHemodynamics,
Griffiths1997TheCondition}. Intuitively, one can think of this outflow boundary condition as
setting the advective flux to being the one consistent with the weak form,
for which there is a unique $c$ satisfying the equations, with $\uu$ determined by the Stokes equations. 

The flux integral over the inflow boundary $\Gin$ cannot be handled
in the same way as the method used for $\Gout$, as this would not preserve
coercivity of the weak form. Lacking experimental data both for the magnitude and
origin of an inflow boundary flux, we deal with the inflow boundary in an explicit
way that resembles an approximation of a periodic boundary condition. At a timestep $n$,
when solving the linear system for the concentration $c^{n+1}$ at the next timestep $n+1$,
we set the inflow flux $h_{\mathrm{in}}^{n+1} = h_{\mathrm{out}}^{n}$, where the latter is the outflow flux
\begin{equation*}
    h_{\mathrm{out}}^{n} = \JJ^n\cdot\nn = (c^n\uu^n - D\nabla c^n)\cdot\nn
\end{equation*}
at the previous timestep. In the limit that the timestep $\Delta t\to 0$, we would be imposing
\begin{equation*}
    \intGin{(\JJ\cdot\nn)} = \intGout{(\JJ\cdot\nn)},
\end{equation*}
which means no mass would be added to the system except for at the injection site,
which is desired as we endeavor to model the ventricles as a closed system.

Using the introduced approaches of handling the in- and outflow boundaries,
the resulting abstract weak formulation for model version II is: find $c\in W$ such that
\begin{equation}
    a(c, \phi) + b(c, \phi) = L(\phi), \quad\forall\phi\in W, \label{eq:adv_diff_weak_formB}
\end{equation}
with the definitions
\begin{align}
    a(c, \phi) &= \intO{\pdifft{c}\phi} - \intO{\JJ\cdot\nabla\phi}, \label{eq:adv_diff_bilinear_form}\\
    b(c, \phi) &= \intGout{c(\uu\cdot\nn)\phi}, \label{eq:adv_diff_skew_form} \\
    L(\phi) &= \intGin{h_{\mathrm{in}}\phi}.\label{eq:adv_diff_linear_form}
\end{align}

\textbf{Baseline model.} Since both free-slip and cilia tangential traction boundaries have the same no-flux boundary condition in the advection-diffusion equation, the weak formulation of the advection-diffusion equation for model version 0 is identical to that of model version II, defined by~\eqref{eq:adv_diff_weak_formB}--\eqref{eq:adv_diff_linear_form}.

\subsection{Discretization of the advection-diffusion equation}
We outline the discretization of the advection-diffusion weak formulation for model version 0, defined by~\eqref{eq:adv_diff_weak_formB}--\eqref{eq:adv_diff_linear_form}. The weak formulation is discretized with a symmetric interior-penalty method using discontinuous Galerkin finite elements~\cite{Arnold1982AnElements}. Let the subscript $h$ denote discrete variables. Exchanging the variables in~\eqref{eq:adv_diff_weak_formB} with their discrete counterparts yields the discrete weak form: find $c_h\in W_h$ such that
\begin{equation*}
    a(c_h, \phi_h) + b(c_h, \phi_h) = L(\phi_h), \quad\forall\phi_h\in W_h.
\end{equation*}
The space $W_h$ is constructed with  finite elements with discontinuous linear Lagrange polynomial basis functions, allowing for discontinuities in the discrete solution across element edges.  To achieve a stable discretization with these elements, we add the stabilization
\begin{equation}
    \begin{aligned}
        a_{\mathrm{stab}}(c_h, \phi_h) = &\sum_{k\in K}\intF{D\frac{\alpha}{\avg{h}}\jump{c_h}\cdot\jump{\phi_h}} +\sum_{k\in K}\intF{\jump{\phi_h}\cdot\jump{c_h \hat{u}_h}}  \\
         -&\sum_{k\in K}\intF{D\avg{\nabla\phi_h}\cdot\jump{c_h}} -\sum_{k\in K}\intF{D\avg{\nabla c_h}\cdot\jump{\phi_h}},
    \end{aligned}\label{eq:SIP_stabilization}
\end{equation}
\mk{
Above and also in weak form of BDM Stokes there should not be the sum over elements.
The sum is over facets. The way it is written you would include each interior facet twice.
}
to the discrete weak formulation, where
\begin{equation*}
    \hat{u}_h = \frac{1}{2}\left(\uu_h\cdot\nn + \vert\uu_h\cdot\nn\vert\right)
\end{equation*}
is used to upwind the velocity $\uu_h$, with the velocity upwinded with respect to a facet $\mathcal{F}_k$. The stabilization parameter $\alpha>0$ must be chosen large enough to achieve stability. The average and jump operators used in~\eqref{eq:SIP_stabilization} are defined, respectively, as
\begin{equation*}
    \avg{c} = \frac{c^+ + c^-}{2} \qquad\mathrm{and}\qquad \jump{c} = c^+\nn^+ + c^-\nn^-.
\end{equation*}
The final discrete weak formulation of the advection-diffusion equation is: find $c_h\in W_h$, such that
\begin{equation}
    a(c_h, \phi_h) + a_{\mathrm{stab}}(c_h, \phi_h) + b(c_h, \phi_h) = L(\phi_h), \quad\forall\phi_h\in W_h.
    \label{eq:discrete_weak_form_adv_diff}
\end{equation}
In this work, we solved~\eqref{eq:discrete_weak_form_adv_diff} numerically with the penalty parameter value $\alpha=25$.

In time, we discretized~\eqref{eq:discrete_weak_form_adv_diff} with the implicit Euler method.

%--------------VERIFICATION--------------%
\section{Verification}\label{sec:appB_verification}
\subsection{Method of Manufactured Solutions}\label{subsec:mms}
Let $\hat{\xx} = (\hat{x}, \hat{y}, \hat{z}) = \frac{1}{L}(x, y, z)$
be the spatial coordinates scaled by the length $L$, which is the length of
the computational domain $\Omega$ in the $x$-direction. Letting $\bm{\phi}$ be the vector-valued function
\begin{equation*}
    \bm{\phi} = \left(\sin{(\pi(\hat{x} - \hat{y}))}, \sin{(\pi(\hat{y} + \hat{z}))}, \sin{(\pi(\hat{x} - \hat{z}))}\right)
\end{equation*}
and defining the velocity as 
\begin{equation}
    \uu_{\mathrm{mms}} = \nabla\times\bm{\phi},
    \label{eq:velocity_mms}
\end{equation}
we have by construction that the velocity is divergence-free,
owing to the identity $\nabla\cdot(\nabla\times\mathbf{F}) = 0$ for
an arbitrary vector field $\mathbf{F}$. The pressure was defined as
\begin{equation}
    p_{\mathrm{mms}} = \cos{(\pi(\hat{x} + \hat{y} + \hat{z}))}.
    \label{eq:pressure_mms}
\end{equation}
Using the Method of Manufactured Solutions~\cite{Roache2001CodeSolutions},
we construct a problem with the Stokes equations where~\eqref{eq:velocity_mms}
and~\eqref{eq:pressure_mms} are the solutions.
To this end, we consider the general form of the steady Stokes equations
\begin{subequations}
    \begin{align}
        -\nabla\cdot\bsig &= \ff \qquad\mathrm{in \ }\Omega,\\
        \nabla\cdot\uu &= g \qquad\mathrm{in \ }\Omega,
    \end{align}
    \label{eq:stokes_mms}%
\end{subequations}%
with $\bsig(\uu, p) = 2\mu\beps(\uu) - p\mathbf{I}$ and
$\beps(\uu) = \frac{1}{2}\left(\nabla\uu + (\nabla\uu)^T\right)$, and the boundary condition
\begin{alignat}{2}
    \uu &= \uu_{\mathrm{mms}} \qquad&&\mathrm{on \ }\Gamma,
\end{alignat}
where $\Gamma=\partial\Omega$ is the boundary of the domain.
Since the velocity field~\eqref{eq:velocity_mms} is divergence free by construction,
$g = 0$. Setting the external force $\ff=-\nabla\cdot\bsig(\uu_{\mathrm{mms}}, p_{\mathrm{mms}})$
using~\eqref{eq:velocity_mms} and~\eqref{eq:pressure_mms},
the solutions $\uu$ and $p$ to~\eqref{eq:stokes_mms} will be the ones
defined by~\eqref{eq:velocity_mms} and~\eqref{eq:pressure_mms},
as long as we enforce equality in the mean pressures, i.e. $\intO{(p_h - p_{\mathrm{mms}})}=0$.
Convergence rates are then readily calculated by comparing the numerical approximations of
$\uu$ and $p$ with $\uu_{\mathrm{mms}}$ and $p_{\mathrm{mms}}$ on meshes of varying grid resolution.

\subsection{Norms and convergence order}\label{subsec:norms_and_eoc}
Errors of the velocity $\uu_h$ and pressure $p_h$ approximated with finite elements
were measured in the $\mathbf{H}^1$ and $L^2$ error norms, respectively.
With the exact velocity field $\uu_{\mathrm{ex}}$, the $\mathbf{H}^1$ error norm was calculated as
\begin{equation*}
    \Hnorm{\uu_h - \uu_{\mathrm{ex}}}^2 = \intO{\nabla(\uu_h - \uu_{\mathrm{ex}}) : \nabla(\uu_h - \uu_{\mathrm{ex}})}.
\end{equation*}
The $L^2$ error norms for the pressure were calculated as
\begin{equation*}
    \normltwo{p_h - p_{\mathrm{ex}}}^2 = \intO{(p_h - p_{\mathrm{ex}})^2}.
\end{equation*}
The approximated functions were interpolated onto elements in a space of piecewise-continuous
Lagrange polynomials of three polynomial degrees higher than the elements used to approximate
the functions, meaning that polynomial degrees 5 and 4 were used for the velocity and pressure
errors, respectively. Numerical integration was performed with a Gauss quadrature rule,
with the quadrature rule degree equal to the polynomial degree of the integrated finite element functions. 

Convergence rates reported in~\Cref{subsec:numerical_verification} were
approximated in the following way. Define the error $e_k$ for a given mesh
resolution $h_k$. Based on the assumption that the error scales as
$e_k \approx C{h_k}^l$, where $C$ is a constant independent of $h_k$,
the convergence order $l$ was approximated as
\begin{equation}
    l = \frac{\log{(e_k/e_{k+1}})}{\log{(h_k/h_{k+1}})}.
    \label{eq:convergence_order_estimate}
\end{equation}
Consider a finite element function 
\begin{equation*}
    \phi = \phi_i\psi_i(\xx)
\end{equation*}
with basis functions $\psi_i$, where $i$ are the degrees of freedom and Einstein summation is employed.
The infinity norm of $\phi$ was approximated from the vector
of coefficients $\left\{\phi_i\right\}$, which represent point evaluations of $\phi$ as
\begin{equation*}
    \normlinf{\phi} = \max_i{\vert \phi_i\vert}.
\end{equation*}

\subsection{Numerical verification of model implementation}\label{subsec:numerical_verification}
A cylinder geometry meshed with varying degrees of refinement was used to
perform convergence analysis of the numerical methods used to solve the Stokes equations.
The cylinder had unit length and a diameter of one half length unit.
For the analysis, we used tangential traction boundary conditions on all
of the cylinder surface. Using the Method of Manufactured Solutions~\cite{Roache2001CodeSolutions}
with the velocity and pressure expressions defined in~\Cref{subsec:mms}
for $\uu_{\mathrm{ex}}$ and $p_{\mathrm{ex}}$, we calculated $\mathbf{H}(\mathrm{div};\Omega)$
and $\mathbf{H}^1$ error norms for $\uu_h$, and the $L^2$ error norm for
$p_h$~(\cref{tab:cylinder_error_rates}). Based on the error norm calculations,
we also calculated the experimental order of convergence~(\Cref{subsec:norms_and_eoc}).
We observe convergence in all of the reported error norms. Furthermore,
we report the error in calculation of the maximum velocities, and observe
convergence in this error as well. Lastly, we calculated the $L^2$ norm of
the divergence $\nabla\cdot\uu_h$ to verify the property of mass conservation.
\begin{table}[!htbp]
    \small
    \centering
    \caption{Error norms of the velocity $\uu_h$ in the $\mathbf{L}^2$ and $\mathbf{H}^1$ norms, the pressure $p_h$ in the $L^2$ norm, and the error in the maximum velocity magnitude $\uu_{h, {\mathrm{mag}}}$ for mesh refinement on a cylinder mesh with model version I. Additionally, we report the $L^2$ norm of the divergence of the velocity. For all error norms, convergence rates calculated with~\eqref{eq:convergence_order_estimate} are given in parentheses.}\label{tab:cylinder_error_rates}
    \begin{tabular}{cc|ccccc}
        \toprule
        $h/h_0$ & \# dofs & $\normltwovec{\uu_h - \uu_{\mathrm{ex}}}$ & $\Hnorm{\uu_h - \uu_{\mathrm{ex}}}$ & $\normltwo{p_h - p_{\mathrm{ex}}}$ & $\normlinf{u_h} -\normlinf{u_{\mathrm{ex}}}$ & $\normltwo{\nabla\cdot\uu_h}$\\ 
        \midrule 
        1    & 8652 & \num{2.3e-02} (---) & \num{4.3e-01} (---) & \num{5.4e-01} (---) & \num{2.4e-02} (---) & \num{1.4e-07} \\  
        1/2  & 41190 & \num{9.3e-03} (1.30) & \num{2.8e-01} (0.61) & \num{3.7e-01} (0.53) & \num{5.0e-03} (2.25) & \num{2.8e-08} \\  
        1/4  & 255016 & \num{3.5e-03} (1.42)    & \num{1.4e-01} (1.00) & \num{2.1e-01} (0.79) & \num{1.4e-03} (1.83) & \num{7.3e-11} \\         
        1/8  & 1855337 & \num{1.3e-03} (1.46) & \num{6.9e-02}  (1.03) & \num{1.1e-01} (0.94)  & \num{6.7e-04} (1.08) & \num{1.4e-10} \\   
        \bottomrule
    \end{tabular}
\end{table}

To verify the model implementation with the zebrafish brain ventricles mesh,
we compared velocity norms for varying degrees of refinement of the brain ventricles mesh~(\Cref{tab:ventricles_norms_BDM}).
For all meshes, we report the maximum discrete velocity magnitude $\normlinf{u_h}$.
For a velocity vector $\uu=(u_x, u_y, u_z)$, the velocity magnitude is
$u = \sqrt{u_x^2 + u_y^2 + u_z^2}$. We also report the maximum value
$\normlinf{u_h}$ scaled by the area $|\Gc|=\intGc{}$ over which tangential traction
$\btau$ was applied. We chose this measure using $|\Gc|$ as a scale because the regions
of the mesh marked as cilia-populated changes slightly with every mesh refinement,
owing to the unstructured organization of the mesh cells. This change in the surface
area over which the tangential traction was applied directly affected the maximum velocity magnitude,
since the maximum velocity magnitude is a result of the total amount of force
$\intGc{\btau\cdot P_{\nn}(\rr)}$ applied on the ciliated boundary, where the cilia
vector $\btau = \tau\lambda(\xx) P_{\nn}(\rr)$ as introduced in~\Cref{subsec:appendixA1}.
In the numerical experiments, we set a constant $\tau = 6.5\times 10^{-4}$ Pa and
$\lambda(\xx)=1$. We set $\rr = (1, 0, 1)$ in the ventral middle ventricle cilia region
and $\rr = -(1, 0, 1)$ in the dorsal middle ventricle and the anterior ventricle cilia
regions. We used $\gamma=10$ for the BDM penalty parameter.

To account for the fact that the volume $|\Omega|=\intO{}$ also changes
slightly with mesh refinement, we report the $\mathbf{L}^2$ norm of $\uu_h$ scaled by $|\Omega|$.
Finally, we provide verification of mass conservation by reporting the $\mathbf{L}^2$ norms of
the divergence $\nabla\cdot\uu_h$. We remark that the units used in the model
implementation was millimeters, which is why~\Cref{tab:ventricles_norms_BDM}
reports units in millimeters, instead of micrometers which has been used elsewhere in the paper. 
\begin{table}[!htbp]
    \centering
    \caption{Velocity norms under mesh refinement on the brain ventricles mesh using model version I with $\btau = \tau P_{\nn}(\rr)$, using $\tau = 6.5\times 10^{-4}$ Pa, $\rr = (1, 0, 1)$ in the ventral middle ventricle and, $\rr = -(1, 0, 1)$ in the dorsal middle ventricle and the anterior ventricle.}\label{tab:ventricles_norms_BDM}
    \begin{tabular}{cc|cccc}
        \toprule
        \# cells & \# dofs & $\normlinf{u_h}$ $\left[\frac{\mathrm{mm}}{\mathrm{s}}\right]$ & $\frac{1}{|\Gc|}\normlinf{u_h}$ $\left[\frac{1}{\mathrm{mm\cdot s}}\right]$ & $\frac{1}{\vert\Omega\vert}\normltwovec{\uu_h}$ $\left[\frac{\mathrm{mm}}{\mathrm{s}}\right]$ & $\normltwo{\nabla\cdot\uu_h}$ $\left[\frac{\mathrm{mm^3}}{\mathrm{s^2}}\right]$\\
        \midrule 
        132134  & 943820  & 0.0276 & 0.97 & \num{3.6e-03} & \num{6.3e-16}\\
        180901  & 1289881 & 0.0284 & 0.98 & \num{3.7e-03} & \num{1.5e-13}\\
        262598  & 1870703 & 0.0292 & 0.98 & \num{3.8e-03} & \num{6.2e-16}\\
        395339  & 2811668 & 0.0292 & 0.96 & \num{3.8e-03} & \num{1.9e-14}\\
        627439  & 4448761 & 0.0286 & 0.92 & \num{3.7e-03} & \num{4.6e-14}\\
        \bottomrule
    \end{tabular}
\end{table}

\section{Quantification of particle transport}\label{sec:appC_particle_transport}
Particle tracking data~\cite{Olstad2019CiliaryDevelopment} was analyzed.
For four zebrafish, five recordings with a few particles for each fish were carried out.
The particles tracked were in the duct between the rhombencephalic ventricle and the
diencephalic ventricle. Frequency of image acquisition was 20.22 Hz.
The collected data constituted $x$ and $y$ positions of the particles at every time
instant of image acquisition. The velocities $\uu_x$ and $\uu_y$ were determined by
calculating the change in positions $\Delta x$ and $\Delta y$ relative to the change
in time $\Delta t$ between two subsequent images, $\Delta t$ being equal to the frequency of acquisition.
Mean velocities $\overline{u}_x$~(\Cref{fig:meanx}) and $\overline{u}_y$~(\Cref{fig:meany}),
which serve as a quantification of particle drift, were calculated
by taking the difference between the last and first positions of a particle.

Particle movement varied over the different measurements. The mean of $\overline{u}_x$ was
0.26 $\mu$m/s and the mean of $\overline{u}_y$ was 0.08 $\mu$m/s. The maximum mean velocities
in absolute values were 10.2 $\mu$m/s and 2.4 $\mu$m/s for $\overline{u}_x$ and $\overline{u}_y$,
respectively. We used flow model II (only cardiac forces) to tune the pressure boundary condition
amplitude $A$. The choice of $A=0.0015$ Pa was used since: (i) it yields a maximum velocity magnitude
of 4.8 $\mu$m/s, which is close to the maximum observed in the particle tracking data when
rejecting the outlier at around 10 $\mu$m/s; (ii) it yields a time-averaged mean velocity
of 0.89 $\mu$m/s, which is close to the mean observed in the particle tracking data.
\begin{figure}[H]
    \begin{minipage}[t]{0.49\textwidth}
        \includegraphics[width=\textwidth]{graphics/mean_velocities_x.png}
        \subcaption{Mean velocities in the $x$ direction of the particles tracked.}\label{fig:meanx}
    \end{minipage}
    \begin{minipage}[t]{0.49\textwidth}
        \includegraphics[width=\textwidth]{graphics/mean_velocities_y.png}
        \subcaption{Mean velocities in the $y$ direction of the particles tracked.}\label{fig:meany}
    \end{minipage} 
    \caption{Particle tracking data. The dashed horizontal line designates a value of zero.}\label{fig:particle_tracking_data}
\end{figure}
\clearpage
\bibliography{ref}{}
\end{document}
